var data =
[
  {
    "data_format": 2,
    "description": "Singularity recipe for bat",
    "filenames": [
      "0.17.1/Singularity"
    ],
    "full_name": "icaoberg/singularity-bat",
    "latest_release": "v0.17.1",
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-bat\" class=\"anchor\" href=\"#singularity-bat\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003esingularity-bat\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"https://camo.githubusercontent.com/7b7c397acc5b91b4c4cf7756015185fe3c5f700f70d256a212de51294a0cf673/68747470733a2f2f696d6775722e636f6d2f724773646e44652e706e67\" target=\"_blank\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/7b7c397acc5b91b4c4cf7756015185fe3c5f700f70d256a212de51294a0cf673/68747470733a2f2f696d6775722e636f6d2f724773646e44652e706e67\" alt=\"Example\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eSingularity recipe for \u003ca href=\"https://github.com/sharkdp/bat\"\u003ebat\u003c/a\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-installing-the-container-on-bridges-2\" class=\"anchor\" href=\"#installing-the-container-on-bridges-2\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstalling the container on Bridges 2\u003c/h2\u003e\n\u003cp\u003eCopy the\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003ccode\u003eSIF\u003c/code\u003e file\u003c/li\u003e\n\u003cli\u003eand the \u003ccode\u003ebat\u003c/code\u003e script\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eto \u003ccode\u003e/opt/packages/bat/0.17.1\u003c/code\u003e.\u003c/p\u003e\n\u003cp\u003eCopy the file \u003ccode\u003emodulefile.lua\u003c/code\u003e to \u003ccode\u003e/opt/modules/bat\u003c/code\u003e as \u003ccode\u003e0.17.1\u003c/code\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-building-the-image-using-the-recipe\" class=\"anchor\" href=\"#building-the-image-using-the-recipe\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBuilding the image using the recipe\u003c/h2\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-build-the-image-locally\" class=\"anchor\" href=\"#to-build-the-image-locally\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo build the image locally\u003c/h3\u003e\n\u003cp\u003eRun the script \u003ccode\u003ebuild.sh\u003c/code\u003e to build image locally\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./build.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-build-the-image-remotely\" class=\"anchor\" href=\"#to-build-the-image-remotely\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo build the image remotely\u003c/h3\u003e\n\u003cp\u003eRun the script \u003ccode\u003erbuild.sh\u003c/code\u003e to build image remotely\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./rbuild.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-to-run-tests\" class=\"anchor\" href=\"#to-run-tests\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo run tests\u003c/h2\u003e\n\u003cp\u003eTo run the available tests, run the command\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./test.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003chr\u003e\n\u003cp\u003eCopyright \u00a9 2020-2021 Pittsburgh Supercomputing Center. All Rights Reserved.\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"http://www.andrew.cmu.edu/~icaoberg\" rel=\"nofollow\"\u003eicaoberg\u003c/a\u003e at the \u003ca href=\"http://www.psc.edu\" rel=\"nofollow\"\u003ePittsburgh Supercomputing Center\u003c/a\u003e in the \u003ca href=\"https://www.cmu.edu/mcs/\" rel=\"nofollow\"\u003eMellon College of Science\u003c/a\u003e at \u003ca href=\"http://www.cmu.edu\" rel=\"nofollow\"\u003eCarnegie Mellon University\u003c/a\u003e.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [
      "singularity",
      "utilities"
    ],
    "updated_at": 1622870361.0
  },
  {
    "data_format": 2,
    "description": "Simulating, Reconstructing and Analysing Data for FEL IDI Experiments",
    "filenames": [
      "Singularity.simple",
      "Singularity.py38",
      "Singularity"
    ],
    "full_name": "fzimmermann89/idi",
    "latest_release": "210601",
    "readme": "\u003cp\u003eCAVE: Hic sunt dracones\u003c/p\u003e\n\u003cp\u003e\u003cem\u003eThe code is a mess, undocumented and only certain code paths are tested.\u003c/em\u003e\u003c/p\u003e\n\u003ch1\u003e\n\u003ca id=\"user-content-idi---incoherent-diffraction-imaging\" class=\"anchor\" href=\"#idi---incoherent-diffraction-imaging\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eIDI - INCOHERENT DIFFRACTION IMAGING\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"https://singularity-hub.org/collections/4824\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a07b23d4880320ac89cdc93bbbb603fa84c215d135e05dd227ba8633a9ff34be/68747470733a2f2f7777772e73696e67756c61726974792d6875622e6f72672f7374617469632f696d672f686f737465642d73696e67756c61726974792d2d6875622d2532336533323932392e737667\" alt=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" data-canonical-src=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"https://github.com/fzimmermann89/idi/actions/workflows/test.yml/badge.svg\" target=\"_blank\" rel=\"noopener noreferrer\"\u003e\u003cimg src=\"https://github.com/fzimmermann89/idi/actions/workflows/test.yml/badge.svg\" alt=\"tests\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"https://camo.githubusercontent.com/a1aa13bc475e383774716a28c54db51e680a438815882cb99e8443eb94a873db/68747470733a2f2f7777772e7472617669732d63692e636f6d2f667a696d6d65726d616e6e38392f6964692e7376673f6272616e63683d6d6173746572\" target=\"_blank\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a1aa13bc475e383774716a28c54db51e680a438815882cb99e8443eb94a873db/68747470733a2f2f7777772e7472617669732d63692e636f6d2f667a696d6d65726d616e6e38392f6964692e7376673f6272616e63683d6d6173746572\" alt=\"Build Status\" data-canonical-src=\"https://www.travis-ci.com/fzimmermann89/idi.svg?branch=master\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eSingularity Image now at \u003ca href=\"https://cloud.sylabs.io/library/_container/607b669a4ad4aa1fdea0c43c\" rel=\"nofollow\"\u003elibrary://fzimmermann89/idi/idi\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eConda Pacakges at \u003ca href=\"https://anaconda.org/zimmf/idi\" rel=\"nofollow\"\u003ezimmf/idi\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003ePIP Source at \u003ca href=\"https://pypi.org/project/idi/\" rel=\"nofollow\"\u003eidi\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eWheels at \u003ca href=\"https://github.com/fzimmermann89/idi/releases/latest\"\u003eReleases\u003c/a\u003e\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-content-of-the-repo\" class=\"anchor\" href=\"#content-of-the-repo\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003econtent of the repo\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003eipynb: example notebooks\u003c/li\u003e\n\u003cli\u003esimulation: simulation of incoherent images\u003c/li\u003e\n\u003cli\u003ereconstruction: direct and ft based reconstruction\u003c/li\u003e\n\u003cli\u003eutil: some small utilities for data analysis, geometry and random distributions, etc.\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-preparation-for-slac-sdf\" class=\"anchor\" href=\"#preparation-for-slac-sdf\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003epreparation for slac sdf:\u003c/h2\u003e\n\u003cp\u003eUse Singulariy, if using OOD launcher, use the following to start a jupyterhub\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003e    function jupyter() { singularity run --app jupyter --nv -B /sdf,/gpfs,/scratch,/lscratch library://fzimmermann89/idi/idi $@; }\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-preparation-for-sacla\" class=\"anchor\" href=\"#preparation-for-sacla\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003epreparation for sacla:\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003eDownload and install miniconda, setup ssh tunnel for web access.\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003econda create -n local3 python=3.7 numpy mkl mkl-dev ipython ipykernel cython jinja2 numba numexpr matplotlib six scipy jupyterlab\u003c/code\u003e\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003econda activate local3\u003c/code\u003e\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003epip install https://github.com/fzimmermann89/idi/\u003c/code\u003e\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003epython -m ipykernel install --user --name local-simulation-env3 --display-name \"local simulation(py37)\"\u003c/code\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003e(C) Felix Zimmermann\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [
      "idi",
      "reconstruction",
      "simulation",
      "xray",
      "incoherent-images",
      "fel"
    ],
    "updated_at": 1622856009.0
  },
  {
    "data_format": 2,
    "description": "Singularity recipe for shellcheck",
    "filenames": [
      "0.5.0/Singularity"
    ],
    "full_name": "icaoberg/singularity-shellcheck",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-naked-singularity\" class=\"anchor\" href=\"#naked-singularity\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003enaked-singularity\u003c/h1\u003e\n\u003cp\u003eA repository of definition files for building\n\u003ca href=\"https://sylabs.io/guides/latest/user-guide\" rel=\"nofollow\"\u003eSingularity\u003c/a\u003e containers\naround the software applications, frameworks, and libraries you need to\nrun on high-performance computing systems.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-install-singularity\" class=\"anchor\" href=\"#install-singularity\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstall Singularity\u003c/h2\u003e\n\u003cp\u003eInstall Singularity on your Linux desktop, laptop, or virtual machine.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003e./naked-singularity.sh install\u003c/pre\u003e\u003c/div\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-build-a-singularity-container-from-a-definition-file\" class=\"anchor\" href=\"#build-a-singularity-container-from-a-definition-file\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBuild a Singularity container from a definition file\u003c/h2\u003e\n\u003cp\u003eBuild an Ubuntu Singularity container from one of the definition files\navailable in this repository.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003esudo singularity build ubuntu.sif definition-files/ubuntu/Singularity.ubuntu-18.04\u003c/pre\u003e\u003c/div\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-download-an-existing-singularity-container\" class=\"anchor\" href=\"#download-an-existing-singularity-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eDownload an existing Singularity container\u003c/h2\u003e\n\u003cp\u003eA number of pre-built containers from this repository are also now\nhosted at Singularity Hub.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003esingularity pull shub://mkandes/naked-singularity:ubuntu-18.04\u003c/pre\u003e\u003c/div\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-status\" class=\"anchor\" href=\"#status\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eStatus\u003c/h2\u003e\n\u003cp\u003eA work in progress. But an important note on the immediate future ...\u003c/p\u003e\n\u003cp\u003eThe future availablity of Singularity Hub is currently unknown and\nmay be slated for retirement as early as April 2021. If and when\nSingularity Hub is retired, the current set of naked-sinularity\ndefinition files will be affected due to the recent move in the past\nyear to using a layered dependency chain of multiple containers built\nand hosted on Singularity Hub. Alternative container build and hosting\noptions are currently under consideration. This upcoming change will\nlikely affect how a given container\u0027s dependency chain is rebuilt and\nsupported long-term.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-contribute\" class=\"anchor\" href=\"#contribute\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eContribute\u003c/h2\u003e\n\u003cp\u003eIf you would like to contribute one of your own Singularity container\ndefinition files for a specific application OR request a modification to\nan existing container definition, then please submit a pull request.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-author\" class=\"anchor\" href=\"#author\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eAuthor\u003c/h2\u003e\n\u003cp\u003eMarty Kandes, Ph.D.\u003cbr\u003e\nComputational \u0026amp; Data Science Research Specialist\u003cbr\u003e\nHigh-Performance Computing User Services Group\u003cbr\u003e\nSan Diego Supercomputer Center\u003cbr\u003e\nUniversity of California, San Diego\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-version\" class=\"anchor\" href=\"#version\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eVersion\u003c/h2\u003e\n\u003cp\u003e1.3.3\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-last-updated\" class=\"anchor\" href=\"#last-updated\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eLast Updated\u003c/h2\u003e\n\u003cp\u003eSaturday, March 13th, 2021\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1622859451.0
  },
  {
    "data_format": 2,
    "description": "A base Singularity container for PRESTO, including dependencies and environment, converted from docker-PRESTO",
    "filenames": [
      "Singularity"
    ],
    "full_name": "federatedcloud/singularity-PRESTO",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-presto\" class=\"anchor\" href=\"#singularity-presto\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003esingularity-PRESTO\u003c/h1\u003e\n\u003cp\u003eA base Singularity container for PRESTO, including dependencies and environment, converted from docker-PRESTO\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"https://singularity-hub.org/collections/4510\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a07b23d4880320ac89cdc93bbbb603fa84c215d135e05dd227ba8633a9ff34be/68747470733a2f2f7777772e73696e67756c61726974792d6875622e6f72672f7374617469632f696d672f686f737465642d73696e67756c61726974792d2d6875622d2532336533323932392e737667\" alt=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" data-canonical-src=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 5,
    "topics": [],
    "updated_at": 1622819998.0
  },
  {
    "data_format": 2,
    "description": "A Docker/Singularity container for packaging pulsar searching software",
    "filenames": [
      "Singularity"
    ],
    "full_name": "federatedcloud/pulsar-pipeline-container",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-docker-pulsar-pipeline\" class=\"anchor\" href=\"#docker-pulsar-pipeline\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003edocker-pulsar-pipeline\u003c/h1\u003e\n\u003cp\u003eA Docker/Singularity container for packaging pulsar searching software\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"https://singularity-hub.org/collections/4541\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a07b23d4880320ac89cdc93bbbb603fa84c215d135e05dd227ba8633a9ff34be/68747470733a2f2f7777772e73696e67756c61726974792d6875622e6f72672f7374617469632f696d672f686f737465642d73696e67756c61726974792d2d6875622d2532336533323932392e737667\" alt=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" data-canonical-src=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 0,
    "topics": [],
    "updated_at": 1622819898.0
  },
  {
    "data_format": 2,
    "description": "A quick reference repository for using the robots in the COR lab",
    "filenames": [
      "containers/singularity/Singularity.ros_melodic-bionic",
      "containers/singularity/Singularity.ros_noetic-focal",
      "real_panda_moveit_control/containers/singularity/Singularity.ros_noetic-focal"
    ],
    "full_name": "rickstaa/COR-robotics-lab-reference",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-cor-robotics-lab-reference\" class=\"anchor\" href=\"#cor-robotics-lab-reference\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eCOR-robotics-lab-reference\u003c/h1\u003e\n\u003cp\u003eA quick reference repository for using the robots in the COR lab. This repository contains several code examples, a \u003ca href=\"https://github.com/rickstaa/COR-robotics-lab-reference/discussions\"\u003ediscussion forum\u003c/a\u003e with FAQ that you might have while working with the robot and a \u003ca href=\"https://github.com/rickstaa/COR-robotics-lab-reference/wiki\"\u003ewiki\u003c/a\u003e with several helpfull documents\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-how-to-reserve-the-robots\" class=\"anchor\" href=\"#how-to-reserve-the-robots\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eHow to reserve the robots\u003c/h2\u003e\n\u003cp\u003ePlease check the \u003cg-emoji class=\"g-emoji\" alias=\"spiral_calendar\" fallback-src=\"https://github.githubassets.com/images/icons/emoji/unicode/1f5d3.png\"\u003e\ud83d\uddd3\ufe0f\u003c/g-emoji\u003e \u003ca href=\"https://github.com/rickstaa/COR-robotics-lab-reference/wiki/%F0%9F%97%93%EF%B8%8F-Robot-reservation-forum\"\u003erobot reservation form\u003c/a\u003e wiki page for more information.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-how-to-work-with-the-robots\" class=\"anchor\" href=\"#how-to-work-with-the-robots\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eHow to work with the robots\u003c/h2\u003e\n\u003cp\u003ePlease check \u003cg-emoji class=\"g-emoji\" alias=\"safety_vest\" fallback-src=\"https://github.githubassets.com/images/icons/emoji/unicode/1f9ba.png\"\u003e\ud83e\uddba\u003c/g-emoji\u003e \u003ca href=\"https://github.com/rickstaa/COR-robotics-lab-reference/wiki/Panda-safety-guidelines\"\u003esafety-guidelines\u003c/a\u003e in the wiki before working with the robot.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1622816661.0
  },
  {
    "data_format": 2,
    "description": "Biobb_structure_utils is the Biobb module collection to modify or extract information from a PDB structure file, such as pulling out a particular model or chain, removing water molecules or ligands, or renumbering or sorting atoms or residues.",
    "filenames": [
      "Singularity.latest"
    ],
    "full_name": "bioexcel/biobb_structure_utils",
    "latest_release": null,
    "readme": "\u003cp\u003e\u003ca href=\"https://biobb-structure-utils.readthedocs.io/en/latest/?badge=latest\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/d7c1b5de86a2921c1f759b175820fb443eba3f18bbf45e56e42f2cee72844627/68747470733a2f2f72656164746865646f63732e6f72672f70726f6a656374732f62696f62622d7374727563747572652d7574696c732f62616467652f3f76657273696f6e3d6c6174657374\" alt=\"\" data-canonical-src=\"https://readthedocs.org/projects/biobb-structure-utils/badge/?version=latest\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"https://anaconda.org/bioconda/biobb_structure_utils\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a477fdb1fd9bc9eb7ffa6cae6a019c6d4c3902fd468b3126f1b78e56c7dcff83/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f696e7374616c6c253230776974682d62696f636f6e64612d627269676874677265656e2e7376673f7374796c653d666c6174\" alt=\"\" data-canonical-src=\"https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"https://quay.io/repository/biocontainers/biobb_structure_utils\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/ca418e4db0b3de91a09a5df4a59446da015b6164598a8bc255918e911484f84f/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f646f636b65722d517561792e696f2d626c7565\" alt=\"\" data-canonical-src=\"https://img.shields.io/badge/docker-Quay.io-blue\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"https://singularity-hub.org/collections/3836\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a07b23d4880320ac89cdc93bbbb603fa84c215d135e05dd227ba8633a9ff34be/68747470733a2f2f7777772e73696e67756c61726974792d6875622e6f72672f7374617469632f696d672f686f737465642d73696e67756c61726974792d2d6875622d2532336533323932392e737667\" alt=\"\" data-canonical-src=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"https://opensource.org/licenses/Apache-2.0\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/2a2157c971b7ae1deb8eb095799440551c33dcf61ea3d965d86b496a5a65df55/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f4c6963656e73652d417061636865253230322e302d626c75652e737667\" alt=\"\" data-canonical-src=\"https://img.shields.io/badge/License-Apache%202.0-blue.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003ch1\u003e\n\u003ca id=\"user-content-biobb_structure_utils\" class=\"anchor\" href=\"#biobb_structure_utils\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003ebiobb_structure_utils\u003c/h1\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-introduction\" class=\"anchor\" href=\"#introduction\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eIntroduction\u003c/h3\u003e\n\u003cp\u003eBiobb_structure_utils is the Biobb module collection to modify or extract information from a PDB structure file, such as pulling out a particular model or chain, removing water molecules or ligands, or renumbering or sorting atoms or residues. Biobb (BioExcel building blocks) packages are Python building blocks that create new layer of compatibility and interoperability over popular bioinformatics tools. The latest documentation of this package can be found in our readthedocs site:\n\u003ca href=\"https://biobb-structure-utils.readthedocs.io/en/latest/\" rel=\"nofollow\"\u003elatest API documentation\u003c/a\u003e.\u003c/p\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-version\" class=\"anchor\" href=\"#version\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eVersion\u003c/h3\u003e\n\u003cp\u003ev3.5.3 2021.2\u003c/p\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-installation\" class=\"anchor\" href=\"#installation\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstallation\u003c/h3\u003e\n\u003cp\u003eUsing PIP:\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003e\u003cstrong\u003eImportant:\u003c/strong\u003e PIP only installs the package. All the dependencies must be installed separately. To perform a complete installation, please use ANACONDA, DOCKER or SINGULARITY.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003eInstallation:\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003e  pip install \"biobb_structure_utils\u0026gt;=3.5.3\"\n\u003c/code\u003e\u003c/pre\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eUsage: \u003ca href=\"https://biobb-structure-utils.readthedocs.io/en/latest/modules.html\" rel=\"nofollow\"\u003ePython API documentation\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eUsing ANACONDA:\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003eInstallation:\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003e  conda install -c bioconda \"biobb_structure_utils\u0026gt;=3.5.3\"\n\u003c/code\u003e\u003c/pre\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eUsage: With conda installation BioBBs can be used with the \u003ca href=\"https://biobb-structure-utils.readthedocs.io/en/latest/modules.html\" rel=\"nofollow\"\u003ePython API documentation\u003c/a\u003e and the \u003ca href=\"https://biobb-structure-utils.readthedocs.io/en/latest/command_line.html\" rel=\"nofollow\"\u003eCommand Line documentation\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eUsing DOCKER:\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003eInstallation:\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003e  docker pull quay.io/biocontainers/biobb_structure_utils:3.5.3--py_0\n\u003c/code\u003e\u003c/pre\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eUsage:\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003e  docker run quay.io/biocontainers/biobb_structure_utils:3.5.3--py_0 \u0026lt;command\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eUsing SINGULARITY:\u003c/p\u003e\n\u003cp\u003e\u003cstrong\u003eMacOS users\u003c/strong\u003e: it\u0027s strongly recommended to avoid Singularity and use \u003cstrong\u003eDocker\u003c/strong\u003e as containerization system.\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003eInstallation:\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003e  singularity pull --name biobb_structure_utils.sif shub://bioexcel/biobb_structure_utils\n\u003c/code\u003e\u003c/pre\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eUsage:\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003e  singularity exec biobb_structure_utils.sif \u0026lt;command\u0026gt;\n\u003c/code\u003e\u003c/pre\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eThe command list and specification can be found at the \u003ca href=\"https://biobb-structure-utils.readthedocs.io/en/latest/command_line.html\" rel=\"nofollow\"\u003eCommand Line documentation\u003c/a\u003e.\u003c/p\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-copyright--licensing\" class=\"anchor\" href=\"#copyright--licensing\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eCopyright \u0026amp; Licensing\u003c/h3\u003e\n\u003cp\u003eThis software has been developed in the \u003ca href=\"http://mmb.irbbarcelona.org\" rel=\"nofollow\"\u003eMMB group\u003c/a\u003e at the \u003ca href=\"http://www.bsc.es/\" rel=\"nofollow\"\u003eBSC\u003c/a\u003e \u0026amp; \u003ca href=\"https://www.irbbarcelona.org/\" rel=\"nofollow\"\u003eIRB\u003c/a\u003e for the \u003ca href=\"http://bioexcel.eu/\" rel=\"nofollow\"\u003eEuropean BioExcel\u003c/a\u003e, funded by the European Commission (EU H2020 \u003ca href=\"http://cordis.europa.eu/projects/823830\" rel=\"nofollow\"\u003e823830\u003c/a\u003e, EU H2020 \u003ca href=\"http://cordis.europa.eu/projects/675728\" rel=\"nofollow\"\u003e675728\u003c/a\u003e).\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e(c) 2015-2020 \u003ca href=\"https://www.bsc.es/\" rel=\"nofollow\"\u003eBarcelona Supercomputing Center\u003c/a\u003e\n\u003c/li\u003e\n\u003cli\u003e(c) 2015-2020 \u003ca href=\"https://www.irbbarcelona.org/\" rel=\"nofollow\"\u003eInstitute for Research in Biomedicine\u003c/a\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eLicensed under the\n\u003ca href=\"https://www.apache.org/licenses/LICENSE-2.0\" rel=\"nofollow\"\u003eApache License 2.0\u003c/a\u003e, see the file LICENSE for details.\u003c/p\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-acknolegements\" class=\"anchor\" href=\"#acknolegements\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eAcknolegements\u003c/h3\u003e\n\u003cp\u003eThis software uses functions to read and modify GRO files based in the \u003ca href=\"https://github.com/caizkun/gropy\"\u003eGROPY\u003c/a\u003e library created by Zhikun Cai (\u003ca href=\"mailto:caizkun@gmail.com\"\u003ecaizkun@gmail.com\u003c/a\u003e) under the \u003ca href=\"https://github.com/caizkun/gropy/blob/master/LICENSE\"\u003eMIT\u003c/a\u003e. In this project \u003ca href=\"https://github.com/caizkun/gropy\"\u003eGROPY\u003c/a\u003e has been adapted to Python 3 and our own needs.\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"https://camo.githubusercontent.com/39c04282e694d49ea0d56c716a27845cd25b9931f791484540f625dcecf68af2/68747470733a2f2f62696f657863656c2e65752f77702d636f6e74656e742f75706c6f6164732f323031392f30342f42696f657863656c6c5f6c6f676f5f3130383070785f7472616e73702e706e67\" target=\"_blank\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/39c04282e694d49ea0d56c716a27845cd25b9931f791484540f625dcecf68af2/68747470733a2f2f62696f657863656c2e65752f77702d636f6e74656e742f75706c6f6164732f323031392f30342f42696f657863656c6c5f6c6f676f5f3130383070785f7472616e73702e706e67\" alt=\"\" title=\"Bioexcel\" data-canonical-src=\"https://bioexcel.eu/wp-content/uploads/2019/04/Bioexcell_logo_1080px_transp.png\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 8,
    "topics": [],
    "updated_at": 1622813374.0
  },
  {
    "data_format": 2,
    "description": null,
    "filenames": [
      "Singularity"
    ],
    "full_name": "timo-singularity/rivet",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-recipes\" class=\"anchor\" href=\"#recipes\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003erecipes\u003c/h1\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1622810530.0
  },
  {
    "data_format": 2,
    "description": "Singularity recipe for graphviz",
    "filenames": [
      "2.44.0/Singularity"
    ],
    "full_name": "icaoberg/singularity-graphviz",
    "latest_release": "v2.44.0",
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-graphviz\" class=\"anchor\" href=\"#singularity-graphviz\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003esingularity-graphviz\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"https://camo.githubusercontent.com/960789693fa68a8f442f9c6cc7d6a117639f1a792ec84c96648ad4764c385fcf/68747470733a2f2f75706c6f61642e77696b696d656469612e6f72672f77696b6970656469612f656e2f342f34382f477261706876697a4c6f676f2e706e67\" target=\"_blank\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/960789693fa68a8f442f9c6cc7d6a117639f1a792ec84c96648ad4764c385fcf/68747470733a2f2f75706c6f61642e77696b696d656469612e6f72672f77696b6970656469612f656e2f342f34382f477261706876697a4c6f676f2e706e67\" alt=\"Logo\" data-canonical-src=\"https://upload.wikimedia.org/wikipedia/en/4/48/GraphvizLogo.png\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eSingularity recipe for \u003ca href=\"https://graphviz.org/\" rel=\"nofollow\"\u003egraphviz \u003c/a\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-installing-the-container-on-bridges-2\" class=\"anchor\" href=\"#installing-the-container-on-bridges-2\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstalling the container on Bridges 2\u003c/h2\u003e\n\u003cp\u003eCopy the\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003ccode\u003eSIF\u003c/code\u003e file\u003c/li\u003e\n\u003cli\u003eand the \u003ccode\u003egraphviz \u003c/code\u003e script\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eto \u003ccode\u003e/opt/packages/graphviz/2.44.0\u003c/code\u003e.\u003c/p\u003e\n\u003cp\u003eCopy the file \u003ccode\u003emodulefile.lua\u003c/code\u003e to \u003ccode\u003e/opt/modules/graphviz \u003c/code\u003e as \u003ccode\u003e 2.44.0\u003c/code\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-building-the-image-using-the-recipe\" class=\"anchor\" href=\"#building-the-image-using-the-recipe\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBuilding the image using the recipe\u003c/h2\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-build-the-image-locally\" class=\"anchor\" href=\"#to-build-the-image-locally\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo build the image locally\u003c/h3\u003e\n\u003cp\u003eRun the script \u003ccode\u003ebuild.sh\u003c/code\u003e to build image locally\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./build.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-build-the-image-remotely\" class=\"anchor\" href=\"#to-build-the-image-remotely\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo build the image remotely\u003c/h3\u003e\n\u003cp\u003eRun the script \u003ccode\u003erbuild.sh\u003c/code\u003e to build image remotely\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./rbuild.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-to-run-tests\" class=\"anchor\" href=\"#to-run-tests\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo run tests\u003c/h2\u003e\n\u003cp\u003eTo run the available tests, run the command\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./test.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003chr\u003e\n\u003cp\u003eCopyright \u00a9 2020-2021 Pittsburgh Supercomputing Center. All Rights Reserved.\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"http://www.andrew.cmu.edu/~icaoberg\" rel=\"nofollow\"\u003eicaoberg\u003c/a\u003e at the \u003ca href=\"http://www.psc.edu\" rel=\"nofollow\"\u003ePittsburgh Supercomputing Center\u003c/a\u003e in the \u003ca href=\"https://www.cmu.edu/mcs/\" rel=\"nofollow\"\u003eMellon College of Science\u003c/a\u003e at \u003ca href=\"http://www.cmu.edu\" rel=\"nofollow\"\u003eCarnegie Mellon University\u003c/a\u003e.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [
      "singularity",
      "utilities"
    ],
    "updated_at": 1622859227.0
  },
  {
    "data_format": 2,
    "description": "Singularity recipe for dust",
    "filenames": [
      "0.5.4/Singularity"
    ],
    "full_name": "icaoberg/singularity-dust",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-dust\" class=\"anchor\" href=\"#singularity-dust\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003esingularity-dust\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"https://github.com/bootandy/dust/raw/master/media/snap.png\" target=\"_blank\" rel=\"noopener noreferrer\"\u003e\u003cimg src=\"https://github.com/bootandy/dust/raw/master/media/snap.png\" alt=\"Example\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eSingularity recipe for \u003ca href=\"https://github.com/bootandy/dust\"\u003edust\u003c/a\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-installing-the-container-on-bridges-2\" class=\"anchor\" href=\"#installing-the-container-on-bridges-2\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstalling the container on Bridges 2\u003c/h2\u003e\n\u003cp\u003eCopy the\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003ccode\u003eSIF\u003c/code\u003e file\u003c/li\u003e\n\u003cli\u003eand the \u003ccode\u003edust\u003c/code\u003e script\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eto \u003ccode\u003e/opt/packages/dust/0.5.4\u003c/code\u003e.\u003c/p\u003e\n\u003cp\u003eCopy the file \u003ccode\u003emodulefile.lua\u003c/code\u003e to \u003ccode\u003e/opt/modulefiles/dust\u003c/code\u003e as \u003ccode\u003e0.5.4\u003c/code\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-building-the-image-using-the-recipe\" class=\"anchor\" href=\"#building-the-image-using-the-recipe\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBuilding the image using the recipe\u003c/h2\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-build-the-image-locally\" class=\"anchor\" href=\"#to-build-the-image-locally\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo build the image locally\u003c/h3\u003e\n\u003cp\u003eRun the script \u003ccode\u003ebuild.sh\u003c/code\u003e to build image locally\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./build.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-build-the-image-remotely\" class=\"anchor\" href=\"#to-build-the-image-remotely\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo build the image remotely\u003c/h3\u003e\n\u003cp\u003eRun the script \u003ccode\u003erbuild.sh\u003c/code\u003e to build image remotely\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./rbuild.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-to-run-tests\" class=\"anchor\" href=\"#to-run-tests\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo run tests\u003c/h2\u003e\n\u003cp\u003eTo run the available tests, run the command\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./test.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003chr\u003e\n\u003cp\u003eCopyright \u00a9 2020-2021 Pittsburgh Supercomputing Center. All Rights Reserved.\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"http://www.andrew.cmu.edu/~icaoberg\" rel=\"nofollow\"\u003eicaoberg\u003c/a\u003e at the \u003ca href=\"http://www.psc.edu\" rel=\"nofollow\"\u003ePittsburgh Supercomputing Center\u003c/a\u003e in the \u003ca href=\"https://www.cmu.edu/mcs/\" rel=\"nofollow\"\u003eMellon College of Science\u003c/a\u003e at \u003ca href=\"http://www.cmu.edu\" rel=\"nofollow\"\u003eCarnegie Mellon University\u003c/a\u003e.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [
      "singularity",
      "utilities"
    ],
    "updated_at": 1622860644.0
  },
  {
    "data_format": 2,
    "description": "Singularity recipe for hyperfine",
    "filenames": [
      "1.11.0/Singularity"
    ],
    "full_name": "icaoberg/singularity-hyperfine",
    "latest_release": "v1.11.0",
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-hyperfine\" class=\"anchor\" href=\"#singularity-hyperfine\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003esingularity-hyperfine\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"https://camo.githubusercontent.com/88a0cb35f42e02e28b0433d4b5e0029e52e723d8feb8df753e1ed06a5161db56/68747470733a2f2f692e696d6775722e636f6d2f7a31394f5978452e676966\" target=\"_blank\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/88a0cb35f42e02e28b0433d4b5e0029e52e723d8feb8df753e1ed06a5161db56/68747470733a2f2f692e696d6775722e636f6d2f7a31394f5978452e676966\" alt=\"Example\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eSingularity recipe for \u003ca href=\"https://github.com/sharkdp/hyperfine\"\u003ehyperfine\u003c/a\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-installing-the-container-on-bridges-2\" class=\"anchor\" href=\"#installing-the-container-on-bridges-2\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstalling the container on Bridges 2\u003c/h2\u003e\n\u003cp\u003eCopy the\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003ccode\u003eSIF\u003c/code\u003e file\u003c/li\u003e\n\u003cli\u003eand the \u003ccode\u003ehyperfine\u003c/code\u003e script\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eto \u003ccode\u003e/opt/packages/hyperfine/1.11.0\u003c/code\u003e.\u003c/p\u003e\n\u003cp\u003eCopy the file \u003ccode\u003emodulefile.lua\u003c/code\u003e to \u003ccode\u003e/opt/modulefiles/hyperfine\u003c/code\u003e as \u003ccode\u003e1.11.0\u003c/code\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-building-the-image-using-the-recipe\" class=\"anchor\" href=\"#building-the-image-using-the-recipe\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBuilding the image using the recipe\u003c/h2\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-build-the-image-locally\" class=\"anchor\" href=\"#to-build-the-image-locally\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo build the image locally\u003c/h3\u003e\n\u003cp\u003eRun the script \u003ccode\u003ebuild.sh\u003c/code\u003e to build image locally\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./build.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-build-the-image-remotely\" class=\"anchor\" href=\"#to-build-the-image-remotely\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo build the image remotely\u003c/h3\u003e\n\u003cp\u003eRun the script \u003ccode\u003erbuild.sh\u003c/code\u003e to build image remotely\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./rbuild.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-to-run-tests\" class=\"anchor\" href=\"#to-run-tests\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo run tests\u003c/h2\u003e\n\u003cp\u003eTo run the available tests, run the command\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./test.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003chr\u003e\n\u003cp\u003eCopyright \u00a9 2020-2021 Pittsburgh Supercomputing Center. All Rights Reserved.\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"http://www.andrew.cmu.edu/~icaoberg\" rel=\"nofollow\"\u003eicaoberg\u003c/a\u003e at the \u003ca href=\"http://www.psc.edu\" rel=\"nofollow\"\u003ePittsburgh Supercomputing Center\u003c/a\u003e in the \u003ca href=\"https://www.cmu.edu/mcs/\" rel=\"nofollow\"\u003eMellon College of Science\u003c/a\u003e at \u003ca href=\"http://www.cmu.edu\" rel=\"nofollow\"\u003eCarnegie Mellon University\u003c/a\u003e.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [
      "singularity",
      "utilities"
    ],
    "updated_at": 1622860679.0
  },
  {
    "data_format": 2,
    "description": "This repository will hold: the build script to install singularity and other dependencies for it, and a definition file for the singularity container for HPC.",
    "filenames": [
      "Singularity"
    ],
    "full_name": "perminaa/SingularityHPC",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularityhpc\" class=\"anchor\" href=\"#singularityhpc\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eSingularityHPC\u003c/h1\u003e\n\u003cp\u003eThis repository will hold: the build script to install singularity and other dependencies for it, and a definition file for the singularity container for HPC.\u003c/p\u003e\n\u003cp\u003eTo install, run \u003ccode\u003egit clone https://github.com/perminaa/SingularityHPC.git \u0026amp;\u0026amp; cd SingularityHPC \u0026amp;\u0026amp; bash buildscript.sh\u003c/code\u003e. This will install and configure singularity\nand build a container called \u003ccode\u003evmdContainer.sif\u003c/code\u003e\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1622857772.0
  },
  {
    "data_format": 2,
    "description": "Singularity recipe for methylpy",
    "filenames": [
      "1.4.3/Singularity"
    ],
    "full_name": "icaoberg/singularity-methylpy",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-bwasp--bisulfite-seq-data-workflow-automation-software-and-protocols\" class=\"anchor\" href=\"#bwasp--bisulfite-seq-data-workflow-automation-software-and-protocols\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBWASP : Bisulfite-seq data Workflow Automation Software and Protocols\u003c/h1\u003e\n\u003cp\u003eThe BWASP repository encompasses code and scripts developed in the\n\u003ca href=\"http://brendelgroup.org/\" rel=\"nofollow\"\u003eBrendel Group\u003c/a\u003e for analyses of bisulfite sequencing\ndata.\nThe entire workflow relies on various other open source software as well as\n\u003ca href=\"https://www.r-project.org/\" rel=\"nofollow\"\u003eR\u003c/a\u003e scripts from the companion\n\u003ca href=\"https://github.com/BrendelGroup/BWASPR\"\u003eBWASPR\u003c/a\u003e repository.\nThe code conforms to our \u003ca href=\"https://brendelgroup.github.io/\" rel=\"nofollow\"\u003eRAMOSE\u003c/a\u003e\nphilosophy: it generates \u003cstrong\u003ereproducible\u003c/strong\u003e, \u003cstrong\u003eaccurate\u003c/strong\u003e, and \u003cstrong\u003emeaningful\u003c/strong\u003e\nresults; it is \u003cstrong\u003eopen\u003c/strong\u003e (source) and designed to be \u003cstrong\u003escalable\u003c/strong\u003e and\n\u003cstrong\u003eeasy\u003c/strong\u003e to use.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-quick-start-\" class=\"anchor\" href=\"#quick-start-\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eQuick Start \u003ca href=\"https://singularity-hub.org/collections/1203\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a07b23d4880320ac89cdc93bbbb603fa84c215d135e05dd227ba8633a9ff34be/68747470733a2f2f7777772e73696e67756c61726974792d6875622e6f72672f7374617469632f696d672f686f737465642d73696e67756c61726974792d2d6875622d2532336533323932392e737667\" alt=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" data-canonical-src=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003c/h2\u003e\n\u003cp\u003eInput to the BWASP workflow consists of accession numbers or fastq files of\nbisulfite-sequencing reads as well as the appropriate genome assembly (and, if\navailable, genome annotation).\nOutput (after read quality control and mapping) are \u003cem\u003e*.mcalls\u003c/em\u003e files that list\nthe sufficiently covered genomic Cs and their methylation percentage in the\ngiven sample.\nThe scripts in the \u003cem\u003ebin\u003c/em\u003e directory take care of minor tasks in the overall\nworkflow, but configuration and execution is via\n\u003ca href=\"https://www.gnu.org/software/make/\" rel=\"nofollow\"\u003eGNU make\u003c/a\u003e using edited copies of the\nmakefiles provided in the \u003cem\u003emakefiles\u003c/em\u003e directory.\nAll the BWASP dependencies are encapsulated in a\n\u003ca href=\"https://www.sylabs.io/docs/\" rel=\"nofollow\"\u003eSingularity\u003c/a\u003e container available from our\n\u003ca href=\"http://BrendelGroup.org/SingularityHub/\" rel=\"nofollow\"\u003eSingularity Hub\u003c/a\u003e.\nThus, once you know what you are doing, execution could be as simple as\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003esingularity pull http://BrendelGroup.org/SingularityHub/bwasp.sif\nsingularity exec bwasp.sif make\n\u003c/code\u003e\u003c/pre\u003e\n\u003cp\u003e(assuming you have prepared a suitable makefile in your working directory).\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-realistic-start\" class=\"anchor\" href=\"#realistic-start\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eRealistic Start\u003c/h2\u003e\n\u003cp\u003ePlease find detailed installation instructions and options in the\n\u003ca href=\"./INSTALL.md\"\u003eINSTALL\u003c/a\u003e document.\nOnce all preparatory steps are taken care of, see the \u003ca href=\"./HOWTO.md\"\u003eHOWTO\u003c/a\u003e\ndocument for a complete example of how to implement and run a workflow.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-reference\" class=\"anchor\" href=\"#reference\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eReference\u003c/h2\u003e\n\u003cp\u003eAmy L. Toth, Murat Ozturk, Saranya Sankaranarayanan, and Volker P. Brendel\n(2018) \u003cem\u003eEstimating the size and dynamics of the CpG methylome of social\ninsects.\u003c/em\u003e To be submitted.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-contact\" class=\"anchor\" href=\"#contact\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eContact\u003c/h2\u003e\n\u003cp\u003ePlease direct all comments and suggestions to\n\u003ca href=\"mailto:vbrendel@indiana.edu\"\u003eVolker Brendel\u003c/a\u003e\nat \u003ca href=\"http://brendelgroup.org/\" rel=\"nofollow\"\u003eIndiana University\u003c/a\u003e.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1622730662.0
  },
  {
    "data_format": 2,
    "description": "Singularity recipe for ABySS",
    "filenames": [
      "2.1.5/Singularity"
    ],
    "full_name": "icaoberg/singularity-abyss",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-bwasp--bisulfite-seq-data-workflow-automation-software-and-protocols\" class=\"anchor\" href=\"#bwasp--bisulfite-seq-data-workflow-automation-software-and-protocols\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBWASP : Bisulfite-seq data Workflow Automation Software and Protocols\u003c/h1\u003e\n\u003cp\u003eThe BWASP repository encompasses code and scripts developed in the\n\u003ca href=\"http://brendelgroup.org/\" rel=\"nofollow\"\u003eBrendel Group\u003c/a\u003e for analyses of bisulfite sequencing\ndata.\nThe entire workflow relies on various other open source software as well as\n\u003ca href=\"https://www.r-project.org/\" rel=\"nofollow\"\u003eR\u003c/a\u003e scripts from the companion\n\u003ca href=\"https://github.com/BrendelGroup/BWASPR\"\u003eBWASPR\u003c/a\u003e repository.\nThe code conforms to our \u003ca href=\"https://brendelgroup.github.io/\" rel=\"nofollow\"\u003eRAMOSE\u003c/a\u003e\nphilosophy: it generates \u003cstrong\u003ereproducible\u003c/strong\u003e, \u003cstrong\u003eaccurate\u003c/strong\u003e, and \u003cstrong\u003emeaningful\u003c/strong\u003e\nresults; it is \u003cstrong\u003eopen\u003c/strong\u003e (source) and designed to be \u003cstrong\u003escalable\u003c/strong\u003e and\n\u003cstrong\u003eeasy\u003c/strong\u003e to use.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-quick-start-\" class=\"anchor\" href=\"#quick-start-\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eQuick Start \u003ca href=\"https://singularity-hub.org/collections/1203\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a07b23d4880320ac89cdc93bbbb603fa84c215d135e05dd227ba8633a9ff34be/68747470733a2f2f7777772e73696e67756c61726974792d6875622e6f72672f7374617469632f696d672f686f737465642d73696e67756c61726974792d2d6875622d2532336533323932392e737667\" alt=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" data-canonical-src=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003c/h2\u003e\n\u003cp\u003eInput to the BWASP workflow consists of accession numbers or fastq files of\nbisulfite-sequencing reads as well as the appropriate genome assembly (and, if\navailable, genome annotation).\nOutput (after read quality control and mapping) are \u003cem\u003e*.mcalls\u003c/em\u003e files that list\nthe sufficiently covered genomic Cs and their methylation percentage in the\ngiven sample.\nThe scripts in the \u003cem\u003ebin\u003c/em\u003e directory take care of minor tasks in the overall\nworkflow, but configuration and execution is via\n\u003ca href=\"https://www.gnu.org/software/make/\" rel=\"nofollow\"\u003eGNU make\u003c/a\u003e using edited copies of the\nmakefiles provided in the \u003cem\u003emakefiles\u003c/em\u003e directory.\nAll the BWASP dependencies are encapsulated in a\n\u003ca href=\"https://www.sylabs.io/docs/\" rel=\"nofollow\"\u003eSingularity\u003c/a\u003e container available from our\n\u003ca href=\"http://BrendelGroup.org/SingularityHub/\" rel=\"nofollow\"\u003eSingularity Hub\u003c/a\u003e.\nThus, once you know what you are doing, execution could be as simple as\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003esingularity pull http://BrendelGroup.org/SingularityHub/bwasp.sif\nsingularity exec bwasp.sif make\n\u003c/code\u003e\u003c/pre\u003e\n\u003cp\u003e(assuming you have prepared a suitable makefile in your working directory).\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-realistic-start\" class=\"anchor\" href=\"#realistic-start\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eRealistic Start\u003c/h2\u003e\n\u003cp\u003ePlease find detailed installation instructions and options in the\n\u003ca href=\"./INSTALL.md\"\u003eINSTALL\u003c/a\u003e document.\nOnce all preparatory steps are taken care of, see the \u003ca href=\"./HOWTO.md\"\u003eHOWTO\u003c/a\u003e\ndocument for a complete example of how to implement and run a workflow.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-reference\" class=\"anchor\" href=\"#reference\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eReference\u003c/h2\u003e\n\u003cp\u003eAmy L. Toth, Murat Ozturk, Saranya Sankaranarayanan, and Volker P. Brendel\n(2018) \u003cem\u003eEstimating the size and dynamics of the CpG methylome of social\ninsects.\u003c/em\u003e To be submitted.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-contact\" class=\"anchor\" href=\"#contact\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eContact\u003c/h2\u003e\n\u003cp\u003ePlease direct all comments and suggestions to\n\u003ca href=\"mailto:vbrendel@indiana.edu\"\u003eVolker Brendel\u003c/a\u003e\nat \u003ca href=\"http://brendelgroup.org/\" rel=\"nofollow\"\u003eIndiana University\u003c/a\u003e.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1622730467.0
  },
  {
    "data_format": 2,
    "description": "Singularity images and recipes",
    "filenames": [
      "clease/Singularity.clease",
      "deal.II/Singularity.deal",
      "graphics/Singularity.gnuplot_5.4a",
      "graphics/Singularity.gnuplot_4.6a",
      "graphics/Singularity.gnuplot_5.4",
      "graphics/Singularity.gnuplot_4.6",
      "graphics/Singularity.graphics",
      "graphics/Singularity.gnuplot_alpine",
      "Atom/Singularity.atom",
      "xcrysden/Singularity.xcrysden_1.5.60",
      "xcrysden/Singularity.xcrysden",
      "ase-twistd/Singularity.ase-twistd",
      "tesseract/Singularity.tesseract",
      "gromacs/Singularity.gromacs",
      "jupyter/Singularity.jupyter",
      "gdis/Singularity.gdis",
      "rstudio-server/Singularity.rstudio-server",
      "mongodb/Singularity.mongodb",
      "MD2-lab/Singularity.md2-lab",
      "pp/Singularity.pp2",
      "tools/Singularity.vim",
      "tools/Singularity.mc",
      "tools/Singularity.gawk",
      "tools/Singularity.gnuplot",
      "tools/Singularity.meld",
      "jmol/Singularity.jmol",
      "ubuntu/Singularity.2004",
      "ubuntu/Singularity.1804",
      "texlive/Singularity.texlive",
      "AMPE/Singularity.ampe",
      "kmos/Singularity.kmos",
      "kmos/Singularity.kmos3_9",
      "emacs/Singularity.emacs",
      "lammps/Singularity.lammps",
      "lammps/Singularity.lammps_ase",
      "lammps/Singularity.lammps_prophet",
      "lammps/Singularity.lammps_ase_kim",
      "cuda/Singularity.u18.04_cuda9.2",
      "VESTA/Singularity.vesta",
      "obabel/Singularity.obabel",
      "acroread/Singularity.acroread"
    ],
    "full_name": "pmitev/Teoroo-singularity",
    "latest_release": null,
    "readme": "\u003cp\u003e\u003ca href=\"https://singularity-hub.org/collections/2338\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a07b23d4880320ac89cdc93bbbb603fa84c215d135e05dd227ba8633a9ff34be/68747470733a2f2f7777772e73696e67756c61726974792d6875622e6f72672f7374617469632f696d672f686f737465642d73696e67756c61726974792d2d6875622d2532336533323932392e737667\" alt=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" data-canonical-src=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003ch1\u003e\n\u003ca id=\"user-content-teoroo-singularity\" class=\"anchor\" href=\"#teoroo-singularity\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTeoroo-singularity\u003c/h1\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [
      "singularity",
      "singularity-containers"
    ],
    "updated_at": 1622723523.0
  },
  {
    "data_format": 2,
    "description": "singularity recipes",
    "filenames": [
      "Singularity.tf2p4_costum",
      "Singularity.tf2p4_addons",
      "Singularity.tf2_addons",
      "Singularity.pandas",
      "Singularity.tf2p1_addons",
      "Singularity.skimage",
      "Singularity.comet",
      "Singularity.tf2p1",
      "Singularity..torch1p8",
      "Singularity.tf2_cuda",
      "Singularity.tf2_cuda_pip",
      "Singularity.tf2",
      "Singularity.torch"
    ],
    "full_name": "xiyaojin/singularity",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity\" class=\"anchor\" href=\"#singularity\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003esingularity\u003c/h1\u003e\n\u003cp\u003esingularity recipes\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 2,
    "topics": [],
    "updated_at": 1622692314.0
  },
  {
    "data_format": 2,
    "description": "RAdiation SEmiconductoR ",
    "filenames": [
      "Singularity"
    ],
    "full_name": "dt-np/raser",
    "latest_release": "v1.0",
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-raser\" class=\"anchor\" href=\"#raser\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eraser\u003c/h1\u003e\n\u003cp\u003eRAdiation SEmiconductoR\u003c/p\u003e\n\u003col\u003e\n\u003cli\u003edocker_build.sh  (.bat for Window)\u003c/li\u003e\n\u003cli\u003edocker_run.sh (.bat for Windows)\u003c/li\u003e\n\u003cli\u003emake (compile raser binary)\u003c/li\u003e\n\u003cli\u003e./run  (run raser program)\u003c/li\u003e\n\u003c/ol\u003e\n\u003ch1\u003e\n\u003ca id=\"user-content-windows-install-software\" class=\"anchor\" href=\"#windows-install-software\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003ewindows install software\u003c/h1\u003e\n\u003cp\u003e1.Docker\u003c/p\u003e\n\u003ch1\u003e\n\u003ca id=\"user-content-add-fenics-in-dockerfile\" class=\"anchor\" href=\"#add-fenics-in-dockerfile\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eadd fenics in dockerfile\u003c/h1\u003e\n\u003cp\u003e2.Xming\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 2,
    "topics": [],
    "updated_at": 1622686281.0
  },
  {
    "data_format": 2,
    "description": "Singularity recipe for GATK",
    "filenames": [
      "4.1.9.0/Singularity"
    ],
    "full_name": "icaoberg/singularity-gatk",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-gatk\" class=\"anchor\" href=\"#singularity-gatk\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003esingularity-gatk\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"https://camo.githubusercontent.com/f0a6e98fde6f4e7dd338612de8a154b1169c4572acc8ca3ffed117a81e28d4be/68747470733a2f2f7468656d652e7a646173736574732e636f6d2f7468656d655f6173736574732f323337383336302f646630383566313534333231666161633931353964646135376635303130336238376134663734332e706e67\" target=\"_blank\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/f0a6e98fde6f4e7dd338612de8a154b1169c4572acc8ca3ffed117a81e28d4be/68747470733a2f2f7468656d652e7a646173736574732e636f6d2f7468656d655f6173736574732f323337383336302f646630383566313534333231666161633931353964646135376635303130336238376134663734332e706e67\" alt=\"Logo\" data-canonical-src=\"https://theme.zdassets.com/theme_assets/2378360/df085f154321faac9159dda57f50103b87a4f743.png\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\nSingularity recipe for \u003ca href=\"https://gatk.broadinstitute.org/hc/en-us\" rel=\"nofollow\"\u003eGATK\u003c/a\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-installing-the-container-on-bridges-2\" class=\"anchor\" href=\"#installing-the-container-on-bridges-2\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstalling the container on Bridges 2\u003c/h2\u003e\n\u003cp\u003eCopy the\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003ccode\u003eSIF\u003c/code\u003e file\u003c/li\u003e\n\u003cli\u003eand the \u003ccode\u003egatk\u003c/code\u003e script\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eto \u003ccode\u003e/opt/packages/gatk/4.1.9.0\u003c/code\u003e.\u003c/p\u003e\n\u003cp\u003eCopy the file \u003ccode\u003emodulefile.lua\u003c/code\u003e to \u003ccode\u003e/opt/modules/gatk\u003c/code\u003e as \u003ccode\u003e4.1.9.0\u003c/code\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-building-the-image-using-the-recipe\" class=\"anchor\" href=\"#building-the-image-using-the-recipe\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBuilding the image using the recipe\u003c/h2\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-build-the-image-locally\" class=\"anchor\" href=\"#to-build-the-image-locally\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo build the image locally\u003c/h3\u003e\n\u003cp\u003eRun the script \u003ccode\u003ebuild.sh\u003c/code\u003e to build image locally.\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./build.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-build-the-image-remotely\" class=\"anchor\" href=\"#to-build-the-image-remotely\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo build the image remotely\u003c/h3\u003e\n\u003cp\u003eRun the script \u003ccode\u003erbuild.sh\u003c/code\u003e to build image locally.\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./rbuild.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-to-run-tests\" class=\"anchor\" href=\"#to-run-tests\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo run tests\u003c/h2\u003e\n\u003cp\u003eTo run the available tests, run the command\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003ebash ./test.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003chr\u003e\n\u003cp\u003eCopyright \u00a9 2020-2021 Pittsburgh Supercomputing Center. All Rights Reserved.\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"http://www.andrew.cmu.edu/~icaoberg\" rel=\"nofollow\"\u003eicaoberg\u003c/a\u003e at the \u003ca href=\"http://www.psc.edu\" rel=\"nofollow\"\u003ePittsburgh Supercomputing Center\u003c/a\u003e in the \u003ca href=\"https://www.cmu.edu/mcs/\" rel=\"nofollow\"\u003eMellon College of Science\u003c/a\u003e at \u003ca href=\"http://www.cmu.edu\" rel=\"nofollow\"\u003eCarnegie Mellon University\u003c/a\u003e.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [
      "singularity",
      "bioinformatics"
    ],
    "updated_at": 1622751567.0
  },
  {
    "data_format": 2,
    "description": "Collection of singularity recipes",
    "filenames": [
      "angsd/Singularity.angsd_v0.933",
      "clumpak/Singularity.clumpak_v1.1",
      "ngsLD/Singularity.ngsLD_v1.1.1",
      "ngsRelate/Singularity.ngsRelate_v2.0",
      "PCAngsd/Singularity.PCAngsd_vlatest",
      "PCAngsd/Singularity.PCAngsd_v0.99"
    ],
    "full_name": "James-S-Santangelo/singularity-recipes",
    "latest_release": null,
    "readme": "\u003cp\u003eThis repository contains Singularity recipes for genomics tools that I have not found available through other means (e.g., Conda, Docker).\u003c/p\u003e\n\u003cp\u003eSingularity images are available on \u003ca href=\"https://cloud.sylabs.io/library/james-s-santangelo\" rel=\"nofollow\"\u003eSylab\u0027s Cloud Library\u003c/a\u003e.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1622647501.0
  },
  {
    "data_format": 2,
    "description": null,
    "filenames": [
      "Singularity",
      "Singularity.salad",
      "Singularity.pokemon"
    ],
    "full_name": "youngchanpark/deploytest",
    "latest_release": "0.0.13.dev1",
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-deploy\" class=\"anchor\" href=\"#singularity-deploy\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eSingularity Deploy\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"img/shpc.png\" target=\"_blank\" rel=\"noopener noreferrer\"\u003e\u003cimg src=\"img/shpc.png\" alt=\"img/shpc.png\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eWouldn\u0027t it be nice to build Singularity images without a registry proper,\nand just keep them alongside the GitHub codebase? This is now possible!\nThis small repository provides an example to get you started. It will\nbuild one or more images (whatever Singularity.* files that are present at\nthe root) and then release them as assets to your GitHub repository so\nthat they can be programatically obtained. It is associated with\n\u003ca href=\"https://github.com/singularityhub/singularity-hpc\"\u003esingularity-hpc\u003c/a\u003e to allow\nyou to then define LMOD modules for these same containers.\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003eCan I upload the largest of chonkers?\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp\u003eYes and no. Note that assets are limited to 2 GB in size, which is still fairly good. You can use\nit as a template for your own recipes as is, or modify it for your custom\nuse case. Instructions are below!\u003c/p\u003e\n\u003cp\u003e\u003cstrong\u003eNote\u003c/strong\u003e Currently recipe extensions are associated with tags, and the GitHub release is\nassociated with a digest. We likely will change this in the next few weeks so that GitHub\nreleases are tags, and the \"digests\" are flie names. Stay tuned for updates!\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-getting-started\" class=\"anchor\" href=\"#getting-started\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eGetting Started\u003c/h2\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-1-template-or-fork\" class=\"anchor\" href=\"#1-template-or-fork\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e1. Template or Fork\u003c/h3\u003e\n\u003cp\u003eIf you haven\u0027t already, template or fork this repository. You can then clone\nyour fork:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003e$ git clone git@github.com:\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003eusername\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e/singularity-deploy\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eYou likely want to name the repository by the container. For example, if I would\nhave created a container on Docker Hub or similar with the name \u003ccode\u003evsoch/salad\u003c/code\u003e,\nhere I\u0027d call the repository \u003ccode\u003esalad\u003c/code\u003e. You obviously are limited to your username\nor an organizational namespace.\u003c/p\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-1-write-your-singularity-recipes\" class=\"anchor\" href=\"#1-write-your-singularity-recipes\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e1. Write your Singularity Recipe(s)\u003c/h3\u003e\n\u003cp\u003eFirst, you should write your container recipe(s) in the present working directory.\nFor good practice, when you are updating recipes you should checkout a new branch\nand open a pull request, as the repository comes with a workflow to trigger on a PR\nto \u003ca href=\".github/workflows/test.yml\"\u003etest your container build\u003c/a\u003e. Note that in the main workflow\nthat deploys the releases, the current branch is set to be \u003ccode\u003emain-branch\u003c/code\u003e. You should\nupdate this to be your main \"production\" branch that you want to deploy releases on merge.\nYou are also free to choose a different trigger and release strategy. You can add any additional\ntests that that you might need. By default, any Singularity.* file will be automatically detected.\nIf there is no extension (the name Singularity), the name used will be \"latest.\"\nYou can use these tags across multiple releases of your containers. For example,\nthese files would generate packages with sifs named as follows:\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003ca href=\"Singularity\"\u003eSingularity\u003c/a\u003e maps to \u003ca href=\"https://github.com/singularityhub/singularity-deploy/releases/download/0.0.1/singularityhub-singularity-deploy.latest.sif\"\u003ehttps://github.com/singularityhub/singularity-deploy/releases/download/0.0.1/singularityhub-singularity-deploy.latest.sif\u003c/a\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"Singularity.pokemon\"\u003eSingularity.pokemon\u003c/a\u003e maps to \u003ca href=\"https://github.com/singularityhub/singularity-deploy/releases/download/0.0.1/singularityhub-singularity-deploy.pokemon.sif\"\u003ehttps://github.com/singularityhub/singularity-deploy/releases/download/0.0.1/singularityhub-singularity-deploy.pokemon.sif\u003c/a\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"Singularity.salad\"\u003eSingularity.salad\u003c/a\u003e maps to \u003ca href=\"https://github.com/singularityhub/singularity-deploy/releases/download/0.0.1/singularityhub-singularity-deploy.latest.sif\"\u003ehttps://github.com/singularityhub/singularity-deploy/releases/download/0.0.1/singularityhub-singularity-deploy.latest.sif\u003c/a\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eFor each name, you can see the direct download URL contains the repository (singularityhub/singularity-deploy),\nYou should not use any \u003ccode\u003e:\u003c/code\u003e characters in either your container tag (the GitHub extension) or\nthe GitHub tags (the release tags) as this might interfere with parsing.\nThe GitHub release tag (0.0.1 in the example above) is discussed next.\u003c/p\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-2-update-the-version-file\" class=\"anchor\" href=\"#2-update-the-version-file\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e2. Update the VERSION file\u003c/h3\u003e\n\u003cp\u003eAny time that you prepare new container recipes, you should update the \u003ca href=\"VERSION\"\u003eVERSION\u003c/a\u003e\nfile. The way that this repository works is to generate a release based on the\nstring in \u003ccode\u003eVERSION\u003c/code\u003e. A version is just a tag, so it could be something like\n\u003ccode\u003e0.0.1\u003c/code\u003e or \u003ccode\u003e0.0.1-slim\u003c/code\u003e. Keep in mind that GitHub releases cannot have duplicated\nnames, so you should not repeat the same tag. Do not use \u003ccode\u003e:\u003c/code\u003e in your tag names.\nIf you do need to re-release a tag (not recommended if a user might be using it and then it\u0027s changed) you can manually delete\nthe release and the tag in the GitHub interface. This is a nice structure because it\nmeans you can have containers with different names under the same tag. In the example\nabove, we have each of \"pokemon,\" \"latest,\" and \"salad\" released under tag 0.0.1.\nThis is how it looks on GitHub:\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"img/releases.png\" target=\"_blank\" rel=\"noopener noreferrer\"\u003e\u003cimg src=\"img/releases.png\" alt=\"img/releases.png\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-3-how-to-develop\" class=\"anchor\" href=\"#3-how-to-develop\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e3. How to Develop\u003c/h3\u003e\n\u003cp\u003eAs we mentioned previously, the container builds will be tested on a pull request,\nand the release will trigger on merge into your main branch (main). See the \u003ca href=\".github/workflows/builder.yml\"\u003e.github/workflows/builder.yml\u003c/a\u003e)\nto edit this. The idea is that you can:\u003c/p\u003e\n\u003col\u003e\n\u003cli\u003eDevelop your container via a development branch\u003c/li\u003e\n\u003cli\u003eOpen a pull request to test the container (the \u003ca href=\".github/workflows/test.yml\"\u003e.github/workflows/test.yml\u003c/a\u003e)\u003c/li\u003e\n\u003cli\u003eOn merge, your container will be released!\u003c/li\u003e\n\u003c/ol\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-4-how-to-pull\" class=\"anchor\" href=\"#4-how-to-pull\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e4. How to pull\u003c/h3\u003e\n\u003cp\u003eTechnically, Singularity can pull just knowing the URL. For example:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003e$ singularity pull https://github.com/singularityhub/singularity-deploy/releases/download/0.0.1/singularityhub-singularity-deploy.latest.sif\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eHowever, the \u003ca href=\"singularity-hpc\"\u003esingularity-hpc\u003c/a\u003e tool (will be) designed to be able to parse and handle\nthese container uris automatically. For the containers here, you could do:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003e$ shpc pull gh://singularityhub/singularity-deploy/0.0.1:latest\n$ shpc pull gh://singularityhub/singularity-deploy/0.0.1:salad\n$ shpc pull gh://singularityhub/singularity-deploy/0.0.1:pokemon\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eor write the container URI into a registry entry:\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003egh: singularityhub/singularity-deploy\nlatest:\n  latest: \"0.0.1\"\ntags:\n  \"latest\": \"0.0.1\"\n  \"salad\": \"0.0.1\"\n  \"pokemon\": \"0.0.1\"\nmaintainer: \"@vsoch\"\nurl: https://github.com/singularityhub/singularity-deploy\n\u003c/code\u003e\u003c/pre\u003e\n\u003cp\u003e(This part is still under development!)\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1622621970.0
  },
  {
    "data_format": 2,
    "description": "A Statistical Framework for Modeling and Identifying Differential Distributions in Single-cell RNA-sequencing Data",
    "filenames": [
      "inst/Singularity"
    ],
    "full_name": "Malindrie/scShapes",
    "latest_release": null,
    "readme": "\n\u003ch1\u003e\n\u003ca id=\"user-content-scshapes\" class=\"anchor\" href=\"#scshapes\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003escShapes\u003c/h1\u003e\n\n\u003cp\u003e\u003ca href=\"https://travis-ci.com/Malindrie/scShapes\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/f46be69a2babf0d92348a4d2abbc1834186998fa8fb4d023c1e85bc5f9405e8a/68747470733a2f2f7472617669732d63692e636f6d2f4d616c696e647269652f73635368617065732e7376673f6272616e63683d6d6173746572\" alt=\"Travis build status\" data-canonical-src=\"https://travis-ci.com/Malindrie/scShapes.svg?branch=master\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"https://ci.appveyor.com/project/Malindrie/scShapes\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/9b16a84bf2877273e52d87887ce418eb0fff59cd1066cca77a8700f6bae73d6b/68747470733a2f2f63692e6170707665796f722e636f6d2f6170692f70726f6a656374732f7374617475732f6769746875622f4d616c696e647269652f73635368617065733f6272616e63683d6d6173746572267376673d74727565\" alt=\"AppVeyor build status\" data-canonical-src=\"https://ci.appveyor.com/api/projects/status/github/Malindrie/scShapes?branch=master\u0026amp;svg=true\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\n\u003cp\u003eWe present a novel statistical framework for identifying differential\ndistributions in single-cell RNA-sequencing (scRNA-seq) data between\ntreatment conditions by modeling gene expression read counts using\ngeneralized linear models (GLMs). We model each gene independently under\neach treatment condition using the error distributions Poisson (P),\nNegative Binomial (NB), Zero-inflated Poisson (ZIP) and Zero-inflated\nNegative Binomial (ZINB) with log link function and model based\nnormalization for differences in sequencing depth. Since all four\ndistributions considered in our framework belong to the same family of\ndistributions, we first perform a Kolmogorov-Smirnov (KS) test to select\ngenes belonging to the family of ZINB distributions. Genes passing the\nKS test will be then modeled using GLMs. Model selection is done by\ncalculating the Bayesian Information Criterion and likelihood ratio test\nstatistic.\u003c/p\u003e\n\u003cp\u003eWhile most methods for differential gene expression analysis aim to\ndetect a shift in the mean of expressed values, single cell data are\ndriven by over-dispersion and dropouts requiring statistical\ndistributions that can handle the excess zeros. By modeling gene\nexpression distributions, our framework can identify subtle variations\nthat do not involve the change in mean. It also has the flexibility to\nadjust for covariates and perform multiple comparisons while explicitly\nmodeling the variability between samples.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-installation\" class=\"anchor\" href=\"#installation\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstallation\u003c/h2\u003e\n\u003cp\u003eYou can install the released version of scShapes with:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-e\"\u003edevtools\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003einstall_github(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\u0027\u003c/span\u003eMalindrie/scShapes\u003cspan class=\"pl-pds\"\u003e\u0027\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003escShapes\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-example\" class=\"anchor\" href=\"#example\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eExample\u003c/h2\u003e\n\u003cp\u003eThis is a basic example which shows how you can use scShapes for\nidentifying differential distributions in single-cell RNA-seq data. For\nthis example data we use the human immune cells (PBMC) dataset\ndistributed through the\n\u003ca href=\"https://github.com/satijalab/seurat-data\"\u003eSeuratData\u003c/a\u003e package.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003escShapes\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eLoading and preparing data for input \u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003eSeurat\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eSeuratData\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eBiocParallel\u003c/span\u003e)\nset.seed(\u003cspan class=\"pl-c1\"\u003e0xBEEF\u003c/span\u003e)\n\nInstallData(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eifnb\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nLoadData(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eifnb\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eifnb.list\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e SplitObject(\u003cspan class=\"pl-smi\"\u003eifnb\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003estim\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eWe first filter the genes to keep only genes expressed in at least 10%\nof cells:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eFirst extract the RNA-seq counts from the \u0027RNA\u0027 assay of the seurat object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eifnb.obj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lapply(\u003cspan class=\"pl-smi\"\u003eifnb.list\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) as.matrix(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eassays\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRNA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lapply(\u003cspan class=\"pl-smi\"\u003eifnb.obj\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) filter_counts(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eperc.zero\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; Removing 527 rows of genes with all zero counts\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u0026gt; Removing 778 rows of genes with all zero counts\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eIn order to normalize for differences in sequencing depth, the log of\nthe total UMI counts assigned per cell will be used as an offset in the\nGLM. This function is inbuilt in the algorithm; however the user is\nrequired to input the library sizes. We can calculate the library sizes\nfor the two treatment conditions as;\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eifnb.lib.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lapply(\u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) apply(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ey\u003c/span\u003e) sum(\u003cspan class=\"pl-smi\"\u003ey\u003c/span\u003e)))\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eThe \u2018meta.data\u2019 slot of the Seurat object also contains information on\nthe cell-types, which will be used as a covariate in the GLM model to\naccount for known biological variation in the data.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eifnb.variables\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lapply(\u003cspan class=\"pl-smi\"\u003eifnb.list\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\n                        \u003cspan class=\"pl-v\"\u003ecell.type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseurat_annotations\u003c/span\u003e),\n                        \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eassays\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRNA\u003c/span\u003e)))\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eFor the purpose of this example we only run the pipeline for randomly\nselected 20 common genes under both treatment conditions \u2018CTRL\u2019 and\n\u2018STIM\u2019.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eRandomly select 20 genes among common genes between the two treatment conditions\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecomm.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e intersect(rownames(\u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e), rownames(\u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003ecomm.20.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e sample(\u003cspan class=\"pl-smi\"\u003ecomm.genes\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereplace\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSubset the randomly selected 20 genes\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eifnb.ctrl\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecomm.20.genes\u003c/span\u003e,]\n\u003cspan class=\"pl-smi\"\u003eifnb.stim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecomm.20.genes\u003c/span\u003e,]\n\u003cspan class=\"pl-smi\"\u003eifnb.subset\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eCTRL\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.ctrl\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eSTIM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.stim\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003ePerform Kolmogorov-Smirnov test to select genes belonging to the family\nof ZINB distributions.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.KS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ks_test(\u003cspan class=\"pl-smi\"\u003eifnb.subset\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexpr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eifnb.variables\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elib.size\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eifnb.lib.size\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eBPPARAM\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eSnowParam(\u003cspan class=\"pl-v\"\u003eworkers\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSOCK\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eifnb.stim.KS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ks_test(\u003cspan class=\"pl-smi\"\u003eifnb.subset\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexpr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eifnb.variables\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elib.size\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eifnb.lib.size\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eBPPARAM\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eSnowParam(\u003cspan class=\"pl-v\"\u003eworkers\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSOCK\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSelect genes significant from the KS test.\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eBy default the \u0027ks_sig\u0027 function performs Benjamini-Hochberg correction for multiple hypothese testing\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eand selects genes significant at p-value of 0.01\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.sig.KS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ks_sig(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.KS\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eifnb.stim.sig.KS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ks_sig(\u003cspan class=\"pl-smi\"\u003eifnb.stim.KS\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSubset UMI counts corresponding to the genes significant from the KS test\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eifnb.sig.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003eCTRL\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e as.data.frame(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.sig.KS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e),\n                       \u003cspan class=\"pl-v\"\u003eSTIM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e as.data.frame(\u003cspan class=\"pl-smi\"\u003eifnb.stim.sig.KS\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.KS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003eifnb.sig.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e),]\n  \u003cspan class=\"pl-smi\"\u003eifnb.stim.KS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003eifnb.filtered\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003eifnb.sig.genes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e),]\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eFit the 4 distributions P,NB,ZIP,ZINB for genes that belong to the ZINB\nfamily of distributions by fitting GLM with log of the library sizes as\nan offset and cell types as a covariate in the GLM.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.fit\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fit_models(\u003cspan class=\"pl-v\"\u003ecounts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.KS\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexpr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eifnb.variables\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elib.size\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eifnb.lib.size\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eBPPARAM\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eSnowParam(\u003cspan class=\"pl-v\"\u003eworkers\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSOCK\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eifnb.stim.fit\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fit_models(\u003cspan class=\"pl-v\"\u003ecounts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eifnb.stim.KS\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecexpr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eifnb.variables\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elib.size\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eifnb.lib.size\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eBPPARAM\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eSnowParam(\u003cspan class=\"pl-v\"\u003eworkers\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSOCK\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eOnce the 4 distributions are fitted, we next calculate the BIC value for\neach model and select the model with the least BIC value.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.bic.val\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e model_bic(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.fit\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eifnb.stim.bic.val\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e model_bic(\u003cspan class=\"pl-smi\"\u003eifnb.stim.fit\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eselect model with least bic value\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.lbic\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lbic_model(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.bic.val\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.ctrl.KS\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eifnb.stim.lbic\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e lbic_model(\u003cspan class=\"pl-smi\"\u003eifnb.stim.bic.val\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.stim.KS\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eTo ensure the fit of the models selected based on the least BIC value,\nadditionally we perform LRT to test for model adequacy and presence of\nzero-inflation.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.gof\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e gof_model(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.lbic\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.variables\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.lib.size\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCTRL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eBPPARAM\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eSerialParam())\n\u003cspan class=\"pl-smi\"\u003eifnb.stim.gof\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e gof_model(\u003cspan class=\"pl-smi\"\u003eifnb.stim.lbic\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.variables\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.lib.size\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSTIM\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eBPPARAM\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eSerialParam())\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eFinally based on the results of the model adequacy tests, we can\nidentify the distribution of best fit for each gene.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e select_model(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.gof\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e select_model(\u003cspan class=\"pl-smi\"\u003eifnb.stim.gof\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eOnce the distribution of best fit is identified for genes of interest,\nit is also possible to extract parameters of interest for the models.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.params\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e model_param (\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.fit\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eifnb.stim.params\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e model_param (\u003cspan class=\"pl-smi\"\u003eifnb.stim.fit\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emodel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eUsing above results we can now identify the differentially distributed\ngenes between \u2018CTRL\u2019 and \u2018STIM\u2019. First we need to subset genes that is\nsignificant in the KS test in both conditions.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSubset the common genes between the two groups, that pass the GOF test\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.fit\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e unlist(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eifnb.stim.fit\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e unlist(\u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eifnb.gof.sig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e intersect(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.fit\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.stim.fit\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eifnb.dist.ctrl\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003egene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eP_genes\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eNB_genes\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZIP_genes\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZINB_genes\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eifnb.dist.ctrl\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edist\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePo\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eP_genes\u003c/span\u003e)), rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eNB_genes\u003c/span\u003e)), rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZIP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZIP_genes\u003c/span\u003e)), rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZINB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eifnb.ctrl.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZINB_genes\u003c/span\u003e)))\n\n\u003cspan class=\"pl-smi\"\u003eifnb.dist.stim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003egene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eP_genes\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eNB_genes\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZIP_genes\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZINB_genes\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eifnb.dist.stim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edist\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePo\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eP_genes\u003c/span\u003e)), rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eNB_genes\u003c/span\u003e)), rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZIP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZIP_genes\u003c/span\u003e)), rep(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eZINB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, length(\u003cspan class=\"pl-smi\"\u003eifnb.stim.dist.fit\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eZINB_genes\u003c/span\u003e)))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eDataframe consisting of distributions followed by each gene passing the KS test\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eifnb.gof.ctrl\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.dist.ctrl\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eifnb.dist.ctrl\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.gof.sig\u003c/span\u003e,]\n\u003cspan class=\"pl-smi\"\u003eifnb.gof.stim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.dist.stim\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eifnb.dist.stim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.gof.sig\u003c/span\u003e,]\n\n\u003cspan class=\"pl-smi\"\u003eifnb.distr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003ectrl\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.gof.ctrl\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edist\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.gof.ctrl\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eifnb.distr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003estim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eifnb.gof.stim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edist\u003c/span\u003e[match(rownames(\u003cspan class=\"pl-smi\"\u003eifnb.distr\u003c/span\u003e), \u003cspan class=\"pl-smi\"\u003eifnb.gof.stim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e)]\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eUsing the dataframe of genes and distribution followed under each\ncondition now we can identify genes changing distribution between \u2018CTRL\u2019\nand \u2018STIM\u2019\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eifnb.DD.genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e change_shape(\u003cspan class=\"pl-smi\"\u003eifnb.distr\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eThis will give a list of two lists with genes changing distribution\nbetween condition and genes changing distribution from unimodal in one\ncondition to zero-inflated in the other condition.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 2,
    "topics": [],
    "updated_at": 1622806325.0
  },
  {
    "data_format": 2,
    "description": null,
    "filenames": [
      "Singularity_fastqc",
      "Singularity_multiqc",
      "Singularity_trimmomatic"
    ],
    "full_name": "uf-icbr-bioinformatics/biocontainers",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-biocontainers\" class=\"anchor\" href=\"#biocontainers\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003ebiocontainers\u003c/h1\u003e\n\u003cp\u003eThis repository contains recipes for containers used to perform QC, summary statistics, and pre-processing on NGS datasets.\u003c/p\u003e\n\u003cp\u003eIn the future, we may provide the containers themselves. Stay tuned. Work in progress.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1622592653.0
  },
  {
    "data_format": 2,
    "description": null,
    "filenames": [
      "container/Singularity"
    ],
    "full_name": "Genomic-Medicine-Linkoping/nextflow_rnaseqfus",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-biocontainers\" class=\"anchor\" href=\"#biocontainers\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003ebiocontainers\u003c/h1\u003e\n\u003cp\u003eThis repository contains recipes for containers used to perform QC, summary statistics, and pre-processing on NGS datasets.\u003c/p\u003e\n\u003cp\u003eIn the future, we may provide the containers themselves. Stay tuned. Work in progress.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 0,
    "topics": [],
    "updated_at": 1622550367.0
  },
  {
    "data_format": 2,
    "description": "Def File of Singularity",
    "filenames": [
      "def/lafin.def",
      "def/wav2pix.def",
      "def/contextual-attention.def",
      "def/singan.def",
      "def/stargan.def",
      "def/edge-connect.def",
      "def/vae-mnist.def",
      "def/sc-fegan.def"
    ],
    "full_name": "Nahuel-Mk2/def-space",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-def-space\" class=\"anchor\" href=\"#def-space\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003edef-space\u003c/h1\u003e\n\u003cp\u003eThis repository is def-space for Singularity\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1606189900.0
  },
  {
    "data_format": 2,
    "description": "My Singularity recipe files",
    "filenames": [
      "bat/Singularity.def",
      "asciinema/Singularity.def",
      "arch-base/Singularity.def",
      "centos-base/Singularity.def",
      "lilypond/Singularity.def",
      "gerda-tgsend/Singularity.def",
      "root-cern/Singularity.def",
      "julia/Singularity.def",
      "itunes/Singularity.def",
      "texlive/Singularity.def"
    ],
    "full_name": "gipert/Singularity.def",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-recipe-files\" class=\"anchor\" href=\"#singularity-recipe-files\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eSingularity recipe files\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"https://github.com/sylabs/singularity\"\u003eSingularity\u003c/a\u003e containers I use the most on HPC clusters.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 2,
    "topics": [
      "singularity",
      "containers"
    ],
    "updated_at": 1587858477.0
  },
  {
    "data_format": 2,
    "description": "Singularity definition files for various projects",
    "filenames": [
      "hauntedhouse_freesurfer/Singularity",
      "hauntedhouse/Singularity",
      "miniconda/Singularity"
    ],
    "full_name": "mvdoc/singularity-def",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-recipe-files\" class=\"anchor\" href=\"#singularity-recipe-files\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eSingularity recipe files\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"https://github.com/sylabs/singularity\"\u003eSingularity\u003c/a\u003e containers I use the most on HPC clusters.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 2,
    "topics": [],
    "updated_at": 1495630055.0
  },
  {
    "data_format": 2,
    "description": "singularity def file for flair(fluka)",
    "filenames": [
      "flair.def",
      "flair-cern.def"
    ],
    "full_name": "ifurther/flair-def",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-recipe-files\" class=\"anchor\" href=\"#singularity-recipe-files\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eSingularity recipe files\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"https://github.com/sylabs/singularity\"\u003eSingularity\u003c/a\u003e containers I use the most on HPC clusters.\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1619686613.0
  },
  {
    "data_format": 2,
    "description": "Singularity definition files for building various software to run on HPC systems",
    "filenames": [
      "checkm.def",
      "instrain.def",
      "octopus.def",
      "torstyverse.def"
    ],
    "full_name": "slhogle/singularity_def_files",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-definition-files\" class=\"anchor\" href=\"#singularity-definition-files\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eSingularity definition files\u003c/h1\u003e\n\u003cp\u003eCollection of def files for building some bioinformatics software I commonly use.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-instrain-v1214\" class=\"anchor\" href=\"#instrain-v1214\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003einStrain v1.2.14\u003c/h2\u003e\n\u003cp\u003e\u003ca href=\"https://github.com/MrOlm/inStrain\"\u003ehttps://github.com/MrOlm/inStrain\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eAlso contains these functioning binaries:\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/samtools/samtools\"\u003esamtools v1.10\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hyattpd/Prodigal\"\u003eprodigal v2.6.3\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/lh3/bwa\"\u003ebwa v0.7.17-r1198-dirty\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/lh3/minimap2\"\u003eminimap2 v2.17 (r941)\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/dnbaker/dashing\"\u003eDashing v0.4.8-1-g47e6\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/ParBLiSS/FastANI\"\u003eFastANI v1.3\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/wwood/CoverM\"\u003eCoverM v0.4.0\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003e\u003ca href=\"https://cloud.sylabs.io/library/slhogle/base/instrain\" rel=\"nofollow\"\u003eImage at Sylabs\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eDownload with:\u003cbr\u003e\n\u003ccode\u003esingularity pull library://slhogle/base/instrain\u003c/code\u003e\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-octopus-development-branch-version-v070-develop-2bde0433\" class=\"anchor\" href=\"#octopus-development-branch-version-v070-develop-2bde0433\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eOctopus development branch version v0.7.0 (develop 2bde0433)\u003c/h2\u003e\n\u003cp\u003e\u003ca href=\"https://github.com/luntergroup/octopus\"\u003ehttps://github.com/luntergroup/octopus\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eBuilt with:\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003epatchelf v0.10\u003c/li\u003e\n\u003cli\u003eopenssl v1.1.1g\u003c/li\u003e\n\u003cli\u003epkg-config v0.29.2\u003c/li\u003e\n\u003cli\u003egpatch v2.7.6\u003c/li\u003e\n\u003cli\u003encurses v6.2\u003c/li\u003e\n\u003cli\u003ecmake v3.17.3\u003c/li\u003e\n\u003cli\u003ehtslib v1.10\u003c/li\u003e\n\u003cli\u003eboost v1.72.0\u003c/li\u003e\n\u003cli\u003eGNU C/C++ compiler v9.3.0\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eTarget: x86_64 Linux 5.3.0-7642-generic\u003cbr\u003e\nSIMD extension: AVX2\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"https://cloud.sylabs.io/library/slhogle/base/octopus\" rel=\"nofollow\"\u003eImage at Sylabs\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eDownload with:\u003cbr\u003e\n\u003ccode\u003esingularity pull library://slhogle/base/octopus\u003c/code\u003e\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-torstyverse\" class=\"anchor\" href=\"#torstyverse\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTorstyverse\u003c/h2\u003e\n\u003cp\u003eBundle of useful packages from \u003ca href=\"https://github.com/tseemann\"\u003eTorsten Seeman\u003c/a\u003e\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/tseemann/samclip\"\u003esampclip v0.4.0\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/tseemann/any2fasta\"\u003eany2fasta v0.4.2\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/tseemann/barrnap\"\u003ebarrnap v0.9\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/tseemann/prokka\"\u003eprokka v1.14.6\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/tseemann/shovill\"\u003eshovill v1.1.0\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/tseemann/abricate\"\u003eabricate v1.0.1\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/tseemann/snippy\"\u003esnippy v4.6.0\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003e\u003ca href=\"https://cloud.sylabs.io/library/slhogle/base/torstyverse\" rel=\"nofollow\"\u003eImage at Sylabs\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eDownload with:\u003cbr\u003e\n\u003ccode\u003esingularity pull library://slhogle/base/torstyverse\u003c/code\u003e\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1614942669.0
  },
  {
    "data_format": 2,
    "description": "Definition (recipe) files for singularity containers.",
    "filenames": [
      "diffnets/diffnets.def",
      "H5N1/Singularity.def",
      "H5N1/Singularity_R_3.6.def",
      "H5N1/h5n1day100.def",
      "cite-seq/Singularity_xenial.def",
      "cite-seq/Singularity_publish.def",
      "cite-seq/Singularity_rocker.def",
      "cite-seq/Singularity_3.def",
      "generic/Singularity.def",
      "bittersweet/Singularity.def",
      "cytof-workflow-v3/Singularity.def",
      "comet/Singularity.def",
      "comet/chemlab-comet.def",
      "cytof-deep-cnn/Singularity.def"
    ],
    "full_name": "rohitfarmer/singularity-defs",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-definitionrecipe-files-for-singularity-containers\" class=\"anchor\" href=\"#definitionrecipe-files-for-singularity-containers\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eDefinition/Recipe Files for Singularity Containers\u003c/h1\u003e\n\u003cp\u003eSome of the containers are available to download from \u003ca href=\"https://cloud.sylabs.io/library/rohitfarmer\" rel=\"nofollow\"\u003ehttps://cloud.sylabs.io/library/rohitfarmer\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eFor feedback and collaboration write to me at \u003ca href=\"mailto:rohit.farmer@gmail.com\"\u003erohit.farmer@gmail.com\u003c/a\u003e\u003c/p\u003e\n\u003ch1\u003e\n\u003ca id=\"user-content-install-singularity-on-linux\" class=\"anchor\" href=\"#install-singularity-on-linux\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstall Singularity on Linux\u003c/h1\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-singularity-version-34\" class=\"anchor\" href=\"#singularity-version-34\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eSingularity Version 3.4\u003c/h2\u003e\n\u003cp\u003eFollow the instructions on \u003ca href=\"https://sylabs.io/guides/3.4/user-guide/quick_start.html#quick-installation-steps\" rel=\"nofollow\"\u003ehttps://sylabs.io/guides/3.4/user-guide/quick_start.html#quick-installation-steps\u003c/a\u003e\u003c/p\u003e\n\u003ch1\u003e\n\u003ca id=\"user-content-building-a-singularity-container\" class=\"anchor\" href=\"#building-a-singularity-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBuilding a Singularity Container\u003c/h1\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-readonly-container\" class=\"anchor\" href=\"#readonly-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eReadonly Container\u003c/h2\u003e\n\u003cp\u003eTo build a read-only SquashFS Singularity container on a local machine using a recipe/definition file.\u003c/p\u003e\n\u003cp\u003e\u003ccode\u003esudo singularity build \u0026lt;container-name.sif\u0026gt; \u0026lt;Singularity.def\u0026gt;\u003c/code\u003e\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-writable-sandbox\" class=\"anchor\" href=\"#writable-sandbox\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eWritable Sandbox\u003c/h2\u003e\n\u003cp\u003eTo build a writable sandbox (essentially a folder) on a local machine using a recipe/definition file.\u003c/p\u003e\n\u003cp\u003e\u003ccode\u003esudo singularity build --sandbox  \u0026lt;sandbox-folder-name/\u0026gt; \u0026lt;Singularity.def\u0026gt;\u003c/code\u003e\u003c/p\u003e\n\u003cp\u003e\u003cem\u003eNote: The advantage of building a writable sandbox is that it can be used to install and configure packages as you go, and once you are satisfied with the requirements, the sandbox can be converted into a read-only SquashFS container. To build a sandbox quickly, it\u0027s better to install a minimal set of packages via the definition file.\u003c/em\u003e\u003c/p\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-installconfigure-packages-in-a-writable-sandbox\" class=\"anchor\" href=\"#installconfigure-packages-in-a-writable-sandbox\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstall/Configure Packages in a Writable Sandbox\u003c/h3\u003e\n\u003cp\u003eOnce a writable sandbox is created to execute it to invoke the shell of the operating installed in the container in the \"writable\" mode. If the shell is not invoked in the \"writable\" mode, all the changes will be lost once you exit from the container environment.\u003c/p\u003e\n\u003cp\u003e\u003ccode\u003esudo singularity shell --writable \u0026lt;sandbox-folder-name/\u0026gt;\u003c/code\u003e\u003c/p\u003e\n\u003cp\u003eInstall packages as you would, for example, in Ubuntu from the command line.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-convert-a-writable-sandbox-to-a-readonly-container\" class=\"anchor\" href=\"#convert-a-writable-sandbox-to-a-readonly-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eConvert a Writable Sandbox to a Readonly Container\u003c/h2\u003e\n\u003cp\u003e\u003ccode\u003esudo singularity build \u0026lt;container-name.sif\u0026gt; \u0026lt;sandbox-folder-name/\u0026gt;\u003c/code\u003e\u003c/p\u003e\n\u003ch1\u003e\n\u003ca id=\"user-content-execute-a-container\" class=\"anchor\" href=\"#execute-a-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eExecute a Container\u003c/h1\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-invoke-a-shell\" class=\"anchor\" href=\"#invoke-a-shell\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInvoke a shell\u003c/h2\u003e\n\u003cp\u003eThe command below can be used for both read-only/writable containers/sandbox.\u003c/p\u003e\n\u003cp\u003e\u003ccode\u003esingularity shell \u0026lt;container-name.sif\u0026gt;\u003c/code\u003e\u003c/p\u003e\n\u003cp\u003e\u003cem\u003eNote: By default, Singularity binds to your current working and home directory. Therefore, you do not need to do anything else to execute a script that is in your current working directory. It can also pull, for example, Vim settings from the .vimrc file in your home directory. Therefore, if Vim installed in the container, it can be used with the same settings from inside the container as it would from outside.\u003c/em\u003e\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-execute-a-command-via-container\" class=\"anchor\" href=\"#execute-a-command-via-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eExecute a Command via Container\u003c/h2\u003e\n\u003cp\u003e\u003ccode\u003esingularity exec \u0026lt;container-name.sif\u0026gt; \u0026lt;command\u0026gt;\u003c/code\u003e\u003c/p\u003e\n\u003cp\u003eFor example: \u003ccode\u003esingularity exec \u0026lt;container-name.sif\u0026gt; Rscript --vanilla hello.R\u003c/code\u003e\u003c/p\u003e\n\u003ch1\u003e\n\u003ca id=\"user-content-running-jupyter-notebooks-from-within-a-container\" class=\"anchor\" href=\"#running-jupyter-notebooks-from-within-a-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eRunning Jupyter Notebooks from Within a Container\u003c/h1\u003e\n\u003cp\u003eThis section is for containers that have Jupyter notebook installed (e.g. cite-seq).\u003c/p\u003e\n\u003cp\u003eA generic command that should work on a personal computer. \u003ccode\u003esingularity exec container-name.sif jupyter notebook --no-browser --ip=127.0.0.1 --port=8888\u003c/code\u003e\u003cbr\u003e\n\u003cem\u003eNote: The IP address and the port number mentioned in the command are the jupyter defaults. They can be changed as per need.\u003c/em\u003e\u003cbr\u003e\nCopy the URL generated by jupyter daemon and paste it in your browser; this should open Jupyter with the list of the files in your current working directory on the host computer.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-running-with-r-as-a-kernel\" class=\"anchor\" href=\"#running-with-r-as-a-kernel\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eRunning with R as a Kernel\u003c/h2\u003e\n\u003cp\u003eSometimes if you already have an R kernel installed in your home directory, it conflicts with what you have inside the container. Therefore, it would require you to re-install the kernel specs in your home directory via the container.\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003esingularity exec container-name.sif R --quiet --slave -e \u0027IRkernel::installspec()\u0027\n\n# Screen log.\n# [InstallKernelSpec] Removing existing kernelspec in /home/user/.local/share/jupyter/kernels/ir\n# [InstallKernelSpec] Installed kernelspec ir in /home/user/.local/share/jupyter/kernels/ir\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-running-on-an-hpc\" class=\"anchor\" href=\"#running-on-an-hpc\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eRunning on an HPC\u003c/h2\u003e\n\u003col\u003e\n\u003cli\u003eSSH to the HPC.\u003c/li\u003e\n\u003cli\u003eClaim an interactive node.\u003c/li\u003e\n\u003cli\u003eNavigate to your project directory. Singularity container should be in your project directory.\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003esingularity exec container-name.sif jupyter notebook --no-browser --ip=\u00270.0.0.0\u0027\u003c/code\u003e\u003c/li\u003e\n\u003cli\u003eKeep the SSH session and Jupyter notebook session running. Copy the URL on your local browser.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp\u003e\u003cem\u003eNote: On some HPCs, you may have to initiate an additional SSH tunnel connecting your local machine to the interactive node on the HPC. In that case, follow some generic instructions here \u003ca href=\"https://rohitfarmer.github.io/docs/docs/HPC/jupyter/\" rel=\"nofollow\"\u003ehttps://rohitfarmer.github.io/docs/docs/HPC/jupyter/\u003c/a\u003e or ask your system administrator.\u003c/em\u003e\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 0,
    "topics": [],
    "updated_at": 1598393438.0
  },
  {
    "data_format": 2,
    "description": "definition files for containers used in Hanlab",
    "filenames": [
      "singularity.SAD/SAD.def",
      "singularity.R.3.6.3.phylo/R.3.6.3.phylo.def",
      "singularity.Rconda/R.3.6.3.def",
      "singularity.R.4.0.2.Bioc/R.4.0.2.Bioc.def",
      "singularity.mkl/mkl.def",
      "singularity.mkl/mkl.ubuntu.def",
      "singularity.py37.ml.openblas/py37.ml.openblas.def",
      "singularity.phylo/phylo.def",
      "singularity.py37.ml.mkl/py37.ml.mkl.def",
      "singularity.R.3.6.3.Bioc/R.3.6.3.Bioc.def",
      "singularity.rnaseq/rnaseq.def"
    ],
    "full_name": "HanLabUNLV/hanlab_singularity_defs",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-hpc_mpi_cuda_singu_def_file\" class=\"anchor\" href=\"#hpc_mpi_cuda_singu_def_file\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003ehpc_mpi_cuda_singu_def_file\u003c/h1\u003e\n\u003cp\u003eA collect of definition files to build images for singularity containers, which includes hpc benchmarks and mpis with cuda support.\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"https://singularity-hub.org/collections/4181\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a07b23d4880320ac89cdc93bbbb603fa84c215d135e05dd227ba8633a9ff34be/68747470733a2f2f7777772e73696e67756c61726974792d6875622e6f72672f7374617469632f696d672f686f737465642d73696e67756c61726974792d2d6875622d2532336533323932392e737667\" alt=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" data-canonical-src=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n",
    "stargazers_count": 0,
    "subscribers_count": 2,
    "topics": [],
    "updated_at": 1617130184.0
  },
  {
    "data_format": 2,
    "description": "Deep phenotyping dashboard",
    "filenames": [
      "singularity/Singularity"
    ],
    "full_name": "dptools/dpdash",
    "latest_release": null,
    "readme": "\u003cp\u003eRead the documentation \u003ca href=\"http://docs.neuroinfo.org/dpdash/en/latest/\" rel=\"nofollow\"\u003ehere\u003c/a\u003e!\u003c/p\u003e\n",
    "stargazers_count": 1,
    "subscribers_count": 2,
    "topics": [],
    "updated_at": 1622821974.0
  },
  {
    "data_format": 2,
    "description": null,
    "filenames": [
      "Singularity.casacore.gpuvmem.9.2",
      "Singularity.casacore.gpuvmem.10.0.ubuntu1604",
      "Singularity",
      "Singularity.casacore.gpuvmem.11.0",
      "Singularity.casacore.gpuvmem.10.0",
      "Singularity.HPC",
      "Singularity.casacore.gpuvmem.9.2.ubuntu1604"
    ],
    "full_name": "miguelcarcamov/container_docker",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-container_docker\" class=\"anchor\" href=\"#container_docker\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003econtainer_docker\u003c/h1\u003e\n\u003cp\u003eUseful containers to work with radio astronomical data\u003c/p\u003e\n\u003cp\u003eMiguel C\u00e1rcamo.\u003c/p\u003e\n",
    "stargazers_count": 1,
    "subscribers_count": 1,
    "topics": [],
    "updated_at": 1622662631.0
  },
  {
    "data_format": 2,
    "description": "To build hpc benchmark and mpi with cuda support sif",
    "filenames": [
      "hpcc_intel.def",
      "bert.def",
      "hpc_mpi_cuda.def",
      "hpl_intel_cuda.def"
    ],
    "full_name": "perambluate/singularity-definition-files-for-HPC",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-hpc_mpi_cuda_singu_def_file\" class=\"anchor\" href=\"#hpc_mpi_cuda_singu_def_file\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003ehpc_mpi_cuda_singu_def_file\u003c/h1\u003e\n\u003cp\u003eA collect of definition files to build images for singularity containers, which includes hpc benchmarks and mpis with cuda support.\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"https://singularity-hub.org/collections/4181\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a07b23d4880320ac89cdc93bbbb603fa84c215d135e05dd227ba8633a9ff34be/68747470733a2f2f7777772e73696e67756c61726974792d6875622e6f72672f7374617469632f696d672f686f737465642d73696e67756c61726974792d2d6875622d2532336533323932392e737667\" alt=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" data-canonical-src=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n",
    "stargazers_count": 1,
    "subscribers_count": 0,
    "topics": [],
    "updated_at": 1588998487.0
  },
  {
    "data_format": 2,
    "description": "Bisulfite-seq data Workflow Automation Software and Protocols",
    "filenames": [
      "Singularity.v0.9",
      "Singularity.v1.1",
      "Singularity",
      "Singularity.v1.0"
    ],
    "full_name": "BrendelGroup/BWASP",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-bwasp--bisulfite-seq-data-workflow-automation-software-and-protocols\" class=\"anchor\" href=\"#bwasp--bisulfite-seq-data-workflow-automation-software-and-protocols\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBWASP : Bisulfite-seq data Workflow Automation Software and Protocols\u003c/h1\u003e\n\u003cp\u003eThe BWASP repository encompasses code and scripts developed in the\n\u003ca href=\"http://brendelgroup.org/\" rel=\"nofollow\"\u003eBrendel Group\u003c/a\u003e for analyses of bisulfite sequencing\ndata.\nThe entire workflow relies on various other open source software as well as\n\u003ca href=\"https://www.r-project.org/\" rel=\"nofollow\"\u003eR\u003c/a\u003e scripts from the companion\n\u003ca href=\"https://github.com/BrendelGroup/BWASPR\"\u003eBWASPR\u003c/a\u003e repository.\nThe code conforms to our \u003ca href=\"https://brendelgroup.github.io/\" rel=\"nofollow\"\u003eRAMOSE\u003c/a\u003e\nphilosophy: it generates \u003cstrong\u003ereproducible\u003c/strong\u003e, \u003cstrong\u003eaccurate\u003c/strong\u003e, and \u003cstrong\u003emeaningful\u003c/strong\u003e\nresults; it is \u003cstrong\u003eopen\u003c/strong\u003e (source) and designed to be \u003cstrong\u003escalable\u003c/strong\u003e and\n\u003cstrong\u003eeasy\u003c/strong\u003e to use.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-quick-start-\" class=\"anchor\" href=\"#quick-start-\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eQuick Start \u003ca href=\"https://singularity-hub.org/collections/1203\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/a07b23d4880320ac89cdc93bbbb603fa84c215d135e05dd227ba8633a9ff34be/68747470733a2f2f7777772e73696e67756c61726974792d6875622e6f72672f7374617469632f696d672f686f737465642d73696e67756c61726974792d2d6875622d2532336533323932392e737667\" alt=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" data-canonical-src=\"https://www.singularity-hub.org/static/img/hosted-singularity--hub-%23e32929.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003c/h2\u003e\n\u003cp\u003eInput to the BWASP workflow consists of accession numbers or fastq files of\nbisulfite-sequencing reads as well as the appropriate genome assembly (and, if\navailable, genome annotation).\nOutput (after read quality control and mapping) are \u003cem\u003e*.mcalls\u003c/em\u003e files that list\nthe sufficiently covered genomic Cs and their methylation percentage in the\ngiven sample.\nThe scripts in the \u003cem\u003ebin\u003c/em\u003e directory take care of minor tasks in the overall\nworkflow, but configuration and execution is via\n\u003ca href=\"https://www.gnu.org/software/make/\" rel=\"nofollow\"\u003eGNU make\u003c/a\u003e using edited copies of the\nmakefiles provided in the \u003cem\u003emakefiles\u003c/em\u003e directory.\nAll the BWASP dependencies are encapsulated in a\n\u003ca href=\"https://www.sylabs.io/docs/\" rel=\"nofollow\"\u003eSingularity\u003c/a\u003e container available from our\n\u003ca href=\"http://BrendelGroup.org/SingularityHub/\" rel=\"nofollow\"\u003eSingularity Hub\u003c/a\u003e.\nThus, once you know what you are doing, execution could be as simple as\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003esingularity pull http://BrendelGroup.org/SingularityHub/bwasp.sif\nsingularity exec bwasp.sif make\n\u003c/code\u003e\u003c/pre\u003e\n\u003cp\u003e(assuming you have prepared a suitable makefile in your working directory).\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-realistic-start\" class=\"anchor\" href=\"#realistic-start\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eRealistic Start\u003c/h2\u003e\n\u003cp\u003ePlease find detailed installation instructions and options in the\n\u003ca href=\"./INSTALL.md\"\u003eINSTALL\u003c/a\u003e document.\nOnce all preparatory steps are taken care of, see the \u003ca href=\"./HOWTO.md\"\u003eHOWTO\u003c/a\u003e\ndocument for a complete example of how to implement and run a workflow.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-reference\" class=\"anchor\" href=\"#reference\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eReference\u003c/h2\u003e\n\u003cp\u003eAmy L. Toth, Murat Ozturk, Saranya Sankaranarayanan, and Volker P. Brendel\n(2018) \u003cem\u003eEstimating the size and dynamics of the CpG methylome of social\ninsects.\u003c/em\u003e To be submitted.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-contact\" class=\"anchor\" href=\"#contact\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eContact\u003c/h2\u003e\n\u003cp\u003ePlease direct all comments and suggestions to\n\u003ca href=\"mailto:vbrendel@indiana.edu\"\u003eVolker Brendel\u003c/a\u003e\nat \u003ca href=\"http://brendelgroup.org/\" rel=\"nofollow\"\u003eIndiana University\u003c/a\u003e.\u003c/p\u003e\n",
    "stargazers_count": 2,
    "subscribers_count": 5,
    "topics": [],
    "updated_at": 1622733671.0
  },
  {
    "data_format": 2,
    "description": "Contains tools for building and distributing the JEDI/JCSDA Containers",
    "filenames": [
      "Singularity.clang-mpich-dev",
      "Singularity.gnu-openmpi-dev",
      "Singularity.tutorial"
    ],
    "full_name": "JCSDA-internal/containers",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-containers\" class=\"anchor\" href=\"#containers\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eContainers\u003c/h1\u003e\n\u003cp\u003eThis repository contains tools for building Singularity and Charliecloud containers.\u003c/p\u003e\n\u003cp\u003eFurthermore, since Intel Docker containers cannot be distributed through Docker Hub, they are also handled here.\u003c/p\u003e\n\u003cp\u003eThe instructions below are intended for the JEDI core team, who are responsible for maintaining JEDI containers and distributing them publicly or privately.\u003c/p\u003e\n\u003cp\u003eHowever, since the JEDI core team cannot legally distribute intel containers for licensing reasons, JEDI users and developers are encouraged to build their own intel development container.\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"myIntel/Intel.md\"\u003eSee here for instructions on how to build your own JEDI Intel development container: Docker, Singularity, or Charliecloud\u003c/a\u003e\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-organization-of-repository\" class=\"anchor\" href=\"#organization-of-repository\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eOrganization of Repository\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003etop-level directory: tools for building Singularity, Docker, and Charliecloud containers\u003c/li\u003e\n\u003cli\u003e\n\u003ccode\u003evagrant\u003c/code\u003e: tools for building Vagrant virtual machines that are provisioned to run JEDI containers\u003c/li\u003e\n\u003cli\u003e\n\u003ccode\u003emodulefiles\u003c/code\u003e, \u003ccode\u003erunscripts\u003c/code\u003e: These directories contain sample modulefiles and batch scripts for running JEDI \"Supercontainers\" across nodes on HPC systems\u003c/li\u003e\n\u003cli\u003e\n\u003ccode\u003emyIntel\u003c/code\u003e is intended to help users from the general JEDI community build their own JEDI intel development containers.\u003c/li\u003e\n\u003cli\u003e\n\u003ccode\u003eintel19\u003c/code\u003e contains deprecated build tools for intel Parallel Studio.\u003c/li\u003e\n\u003cli\u003e\n\u003ccode\u003eexamples\u003c/code\u003e is a sandbox, containing instructive examples of how to implement features that may not be used now but might be used in the future.  An example is how to build writable singularity containers.   These scripts are not maintained; there is no guarantee that they will run as is.\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-prerequisites\" class=\"anchor\" href=\"#prerequisites\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003ePrerequisites\u003c/h2\u003e\n\u003cp\u003eIn order to build Docker, Singularity, or Charliecloud containers, you will of course need to have the appropriate software installed, namely \u003ccode\u003edocker\u003c/code\u003e, \u003ccode\u003esingularity\u003c/code\u003e, or \u003ccode\u003echarliecloud\u003c/code\u003e.  Members of the JEDI core team can launch an AWS node with all of these pre-installed.  Or, you can install them yourself as described in the JEDI documentation.\u003c/p\u003e\n\u003cp\u003eThe scripts in this directory also assume that you have root privileges.\u003c/p\u003e\n\u003cp\u003eAlso, core developers often find it necessary to access feature or bugfix branches of the jedi stack for testing purposes.  So, the \u003ccode\u003ebuild_container.sh\u003c/code\u003e script uses the JCSDA-internal (private) jedi-stack repo.  For this reason, you need to provide an ssh key for access.  This script uses a generic academy ssh key to ensure that it has read-only access to selected JCSDA repositories.  If you do not have access to this key, you can replace it with another by changing the \u003ccode\u003eKEY\u003c/code\u003e variable in \u003ccode\u003ebuild_containers.sh\u003c/code\u003e.  But it is recommended to retain the read-only access.  You can build the \u003ccode\u003emyIntel\u003c/code\u003e container without an ssh key.\u003c/p\u003e\n\u003cp\u003eNote: to build the tutorial container you have to copy the ssh key into the directory \u003ccode\u003essh-key\u003c/code\u003e and modify the singularity recipe file accordingly if it has a different name.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-build-a-dev-container\" class=\"anchor\" href=\"#build-a-dev-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBuild a dev container\u003c/h2\u003e\n\u003cp\u003eTo build a Singularity, Charliecloud, and/or a Docker container, enter this and respond to the prompts to build the containers of your choice.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003e./build_containers.sh \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003ename\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003ewhere \u003ccode\u003e\u0026lt;name\u0026gt;\u003c/code\u003e matches one of the available Dockerfile extensions, e.g. \u003ccode\u003egnu-openmpi-dev\u003c/code\u003e.  It also accepts an optional second argument to specify a tag.  The default tag is \u003ccode\u003ebeta\u003c/code\u003e.\u003c/p\u003e\n\u003cp\u003eFor the the gnu and clang containers, the Singularity containers are built directly from the images on Docker Hub.  A Docker container will only be created if you choose to build a Charliecloud container, which is then built from the Docker container.\u003c/p\u003e\n\u003cp\u003eFor the \u003ccode\u003eintel-impi-dev\u003c/code\u003e container, a Docker file is always created and then the Singularity and Charliecloud containers are created from that.\u003c/p\u003e\n\u003cp\u003eThe intel Docker container is the one used for CI so it is kept relatively compact.  If you wish to add additional components such as Vtune, it is recommended you use the companion scripts in the \u003ccode\u003emyIntel\u003c/code\u003e directory.  These scripts are simplified in the root directory but they are intended for use by the general JEDI user and developer community.  The main simplification is that there is no need to supply an ssh key because those scripts only access the public jedi-stack repo.\u003c/p\u003e\n\u003cp\u003eTo build the tutorial container, just specify \u003ccode\u003etutorial\u003c/code\u003e for the \u003ccode\u003e\u0026lt;name\u0026gt;\u003c/code\u003e.  The tutorial container is exclusively a Singularity container and uses GNU-OpenMPI: There are no clang or intel options and there are no Docker or Charliecloud containers created.\u003c/p\u003e\n\u003cp\u003eThe Singularity and Charliecloud container files will be placed in a subdirectory called \u003ccode\u003econtainers\u003c/code\u003e.\u003c/p\u003e\n\u003cp\u003eNote: building the Mellanox-enabled HPC container isn\u0027t yet automated.  For this or other non-standard cases, you can edit the Dockerfiles, Singularity files, and scripts manually as needed.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-test-the-container\" class=\"anchor\" href=\"#test-the-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTest the container\u003c/h2\u003e\n\u003cp\u003eBefore distributing a container, it\u0027s always important to test it.  A good test is usually to enter the container and then build and test fv3-bundle.\u003c/p\u003e\n\u003cp\u003eTo enter the Singularity container, enter:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003esingularity shell -e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003ename\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e.sif\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eAnd, for CharlieCloud, you can do this:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003emkdir -p \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/ch-jedi\n\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/ch-jedi\nch-tar2dir \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003epath-to-tarfile\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e/ch-jedi-gnu-openmpi-dev.tar.gz \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\nch-run ch-jedi-gnu-openmpi-dev -- bash\u003c/pre\u003e\u003c/div\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-distribute-the-latest-container\" class=\"anchor\" href=\"#distribute-the-latest-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eDistribute the latest container\u003c/h2\u003e\n\u003cp\u003eThe latest Singularity containers are made available on Sylabs Cloud, the latest Charliecloud containers are made available on a public AWS S3 bucket, and the latest intel containers are made available on a private AWS S3 bucket.  The purpose of the \u003ccode\u003epush_containers.sh\u003c/code\u003e script is to push the new container to these distribution sites.\u003c/p\u003e\n\u003cp\u003eThe \u003ccode\u003ebeta\u003c/code\u003e tag is a special case.  If the tag is \u003ccode\u003ebeta\u003c/code\u003e, it is assumed that, after it passes tests, this container is ready to be deployed as \u003ccode\u003elatest\u003c/code\u003e.  In this case, a copy of the current \u003ccode\u003elatest\u003c/code\u003e container is saved with the tag \u003ccode\u003erevert\u003c/code\u003e.\u003c/p\u003e\n\u003cp\u003eSo, the typical workflow would be to enter\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003e./push_containers.sh \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003ename\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eAs with \u003ccode\u003ebuild_containers.sh\u003c/code\u003e, \u003ccode\u003epush_containers.sh\u003c/code\u003e accepts an optional second argument which is a tag.  This is sometimes useful for experimental cases but is not part of the normal workflow.\u003c/p\u003e\n\u003cp\u003eFor instructions on how to download these containers, see \u003ca href=\"https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/using/jedi_environment/containers.html#available-containers\" rel=\"nofollow\"\u003ethe JEDI Documentation\u003c/a\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-tagged-releases\" class=\"anchor\" href=\"#tagged-releases\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTagged Releases\u003c/h2\u003e\n\u003cp\u003eMost developers use the latest development containers but it\u0027s also useful to have tagged containers that accompany JEDI releases.  This is particularly relevant for scientific users (as opposed to developers) who may wish to use tagged releases and containers for reproducibility in research.  Tagged containers can also be used to provide stability for operational or Near-Real-Time (NRT) workflows.\u003c/p\u003e\n\u003cp\u003eSylabs cloud has a storage quota (currently 11 GB) that would be quickly overwhelmed if we were to store many release containers there.  So, this is reserved for \"latest\" and \"revert\".\u003c/p\u003e\n\u003cp\u003eTagged singularity containers are distributed on the \u003ca href=\"http://data.jcsda.org/pages/containers.html\" rel=\"nofollow\"\u003eJCSDA Public Container Repository\u003c/a\u003e along with the latest and tagged Charliecloud containers.  Tagged docker release containers can be obtained from Docker Hub.  For example:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003edocker pull jcsda/docker-gnu-openmpi-dev:v1.0.0\u003c/pre\u003e\u003c/div\u003e\n",
    "stargazers_count": 2,
    "subscribers_count": 5,
    "topics": [],
    "updated_at": 1622674490.0
  },
  {
    "data_format": 2,
    "description": "A collection of scripts to run variant aggregation tests from whole-genome sequencing data.",
    "filenames": [
      "Singularity_via_docker",
      "Singularity"
    ],
    "full_name": "hmgu-itg/burden_testing",
    "latest_release": "v1.0",
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-mummy-the-wrapped-monster\" class=\"anchor\" href=\"#mummy-the-wrapped-monster\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eMummy (the wrapped MONSTER)\u003c/h1\u003e\n\u003cp\u003eThis is a pipeline to run genome-wide burdent tests using sequencing data. Head over to \u003ca href=\"https://github.com/hmgu-itg/burden_testing/wiki\"\u003ethe wiki\u003c/a\u003e for detailed instructions on how to run it.\u003c/p\u003e\n",
    "stargazers_count": 2,
    "subscribers_count": 2,
    "topics": [],
    "updated_at": 1622649802.0
  },
  {
    "data_format": 2,
    "description": "definition files and wrapper scripts used by NIH HPC staff to install user-facing apps on the Biowulf cluster",
    "filenames": [
      "molecular-modeling-graphics/blender/2.82/blender.def",
      "molecular-modeling-graphics/starseqr/0.6.7/starseqr.def",
      "molecular-modeling-graphics/chimerax/1.1/chimerax.def",
      "molecular-modeling-graphics/chimerax/0.93/chimerax.def",
      "systems-biology/cellphonedb/2.1.7/cellphonedb.def",
      "systems-biology/cellphonedb/2.1.2/cellphonedb.def",
      "sequence-analysis/cicero/1.4.0/cicero.def",
      "sequence-analysis/roary/3.13.0/roary.def",
      "sequence-analysis/roary/3.12.0/roary.def",
      "sequence-analysis/arriba/2.0.0/arriba.def",
      "sequence-analysis/arriba/1.2.0/arriba.def",
      "sequence-analysis/xhla/2018-04-04/xhla.def",
      "sequence-analysis/qtltools/1.3.1/qtltools.def",
      "sequence-analysis/glu/1.0b3/glu.def",
      "sequence-analysis/annogesic/1.0.2/annogesic.def",
      "sequence-analysis/focus/0.6.10/focus.def",
      "sequence-analysis/wisexome/20180814/wisexome.def",
      "sequence-analysis/anvio/7/anvio.def",
      "sequence-analysis/cactus/1.2.3/cactus.def",
      "sequence-analysis/bamgineer/2-20200624/bamgineer.def",
      "sequence-analysis/svtools/0.5.1/svtools.def",
      "sequence-analysis/phaser/1.1.1/phaser.def",
      "sequence-analysis/accurity/20210209/accurity.def",
      "sequence-analysis/accurity/20180724/accurity.def",
      "sequence-analysis/sonicparanoid/1.3.5/sonicparanoid.def",
      "sequence-analysis/sonicparanoid/1.3.2/sonicparanoid.def",
      "sequence-analysis/orffinder/0.4.3-sing-install/orffinder.def",
      "sequence-analysis/vcf-kit/0.1.6/vcf-kit.def",
      "sequence-analysis/smoove/0.2.5/smoove.def",
      "sequence-analysis/smoove/0.2.1/smoove.def",
      "sequence-analysis/m-tools/20210208/m-tools.def",
      "sequence-analysis/eukrep/20180308/eukrep.def",
      "sequence-analysis/asgal/1.0/asgal.def",
      "sequence-analysis/netoglyc/3.1d/netoglyc.def",
      "sequence-analysis/saige/0.44.1/saige.def",
      "sequence-analysis/intarna/3.2.0/intarna.def",
      "sequence-analysis/chipseq_pipeline/1.2.0/chipseq_pipeline.def",
      "sequence-analysis/mmarge/1.0/mmarge.def",
      "sequence-analysis/acfs/20180316/acfs.def",
      "sequence-analysis/ldsc/3d0c4464/ldsc.def",
      "sequence-analysis/deepsea/0.94c/deepsea.def",
      "sequence-analysis/braker/2/braker.def",
      "sequence-analysis/prokka/1.13/prokka.def",
      "sequence-analysis/prokka/1.14.6/prokka.def",
      "sequence-analysis/htgtsrep/9fe74ff/htgtsrep.def",
      "sequence-analysis/glnexus/1.1.11/glnexus.def",
      "sequence-analysis/glnexus/1.2.7/glnexus.def",
      "sequence-analysis/augustus/3.3.3/augustus.def",
      "image-analysis/terastitcher/1.11.10/terastitcher.def",
      "image-analysis/terastitcher/1.10.8/terastitcher.def",
      "image-analysis/mrtrix/3.0.2-cuda9.1/mrtrix.def",
      "image-analysis/mrtrix/3.0.1/mrtrix.def",
      "image-analysis/mrtrix/3.0.0/mrtrix.def",
      "image-analysis/mriqc/0.16.1/mriqc.def",
      "image-analysis/mriqc/0.15.1/mriqc.def",
      "image-analysis/mriqc/0.15.2/mriqc.def",
      "image-analysis/mriqc/0.15.2-0be03bf/mriqc.def",
      "image-analysis/xcpengine/1.2.1/xcpengine.def",
      "image-analysis/xcpengine/1.0/xcpengine.def",
      "image-analysis/xcpengine/1.2.3/xcpengine.def",
      "image-analysis/tesseract/4.1.1/tesseract.def",
      "image-analysis/civet/2.1.1/civet.def",
      "image-analysis/topaz/0.2.5/topaz.def",
      "image-analysis/deepmedic/0.8.0/deepmedic.def",
      "image-analysis/deepmedic/0.8.2/deepmedic.def",
      "image-analysis/broccoli/1.0.1/broccoli.def",
      "image-analysis/qsiprep/0.8.0/qsiprep.def",
      "image-analysis/fitlins/0.8.0/fitlins.def",
      "image-analysis/fitlins/0.7.0/fitlins.def",
      "image-analysis/minc-toolkit/1.9.16/minc-toolkit.def",
      "image-analysis/minc-toolkit/1.9.18/minc-toolkit.def",
      "image-analysis/resmap/1.95/resmap.def",
      "image-analysis/fmriprep/20.2.1/fmriprep.def",
      "image-analysis/fmriprep/20.0.5/fmriprep.def",
      "image-analysis/fmriprep/20.1.1/fmriprep.def",
      "image-analysis/fmriprep/20.1.3/fmriprep.def",
      "image-analysis/fmriprep/20.2.0/fmriprep.def",
      "image-analysis/baracus/1.1.4/baracus.def",
      "computational-chemistry/ampl/f35623d4/ampl.def",
      "mass-spectrometry/maxquant/1.6.17.0/maxquant.def",
      "mass-spectrometry/maxquant/1.6.3.3/maxquant.def",
      "mass-spectrometry/maxquant/1.6.7.0/maxquant.def",
      "utilities/sysbench/1.0.11/sysbench.def",
      "utilities/sysbench/1.0.20/sysbench.def",
      "utilities/uropa/3.5.0/uropa.def",
      "utilities/pyega3/3.3.0/pyega3.def",
      "utilities/whatshap/0.18/whatshap.def",
      "utilities/snp-sites/2.4.1/snp-sites.def",
      "utilities/gdc-client/1.5.0/gdc-client.def",
      "utilities/datalad/0.13.0rc2/datalad.def",
      "utilities/visidata/2.2/visidata.def",
      "utilities/ariba/2.14.4/ariba.def",
      "utilities/atom/1.13.1/atom.def",
      "utilities/xvfb/1.19.6/xvfb.def",
      "utilities/longshot/0.3.5/longshot.def",
      "utilities/pdf2svg/0.2.3/pdf2svg.def",
      "utilities/vcf2db/2020.09.14/vcf2db.def",
      "high-throughput-sequencing/pepr/1.1.24/pepr.def",
      "high-throughput-sequencing/cicero/0.3.0/cicero.def",
      "high-throughput-sequencing/deepsignal/0.1.8/deepsignal.def",
      "high-throughput-sequencing/bamutil/1.0.15/bamutil.def",
      "high-throughput-sequencing/vagrent/3.3.4/vagrent.def",
      "high-throughput-sequencing/metaphlan/3.0/metaphlan.def",
      "high-throughput-sequencing/metaphlan/3.0.6/metaphlan.def",
      "high-throughput-sequencing/hap.py/0.3.9/hap.py.def",
      "high-throughput-sequencing/rilseq/0.75/rilseq.def",
      "high-throughput-sequencing/tetoolkit/2.2.1/tetoolkit.def",
      "high-throughput-sequencing/tetoolkit/2.1.4/tetoolkit.def",
      "high-throughput-sequencing/ascatngs/4.3.4/ascatngs.def",
      "high-throughput-sequencing/ascatngs/4.3.3/ascatngs.def",
      "high-throughput-sequencing/ascatngs/4.5.0/ascatngs.def",
      "high-throughput-sequencing/htseq/0.11.4/htseq.def",
      "high-throughput-sequencing/metabat/2.13/metabat.def",
      "high-throughput-sequencing/atropos/1.1.18/atropos.def",
      "high-throughput-sequencing/epic2/0.0.41/epic2.def",
      "high-throughput-sequencing/deeptools/3.4.2/deeptools.def",
      "high-throughput-sequencing/deeptools/3.5.0/deeptools.def",
      "high-throughput-sequencing/vep/101/vep.def",
      "high-throughput-sequencing/vep/103/vep.def",
      "high-throughput-sequencing/vep/97/vep.def",
      "high-throughput-sequencing/atac_dnase_pipelines/0.3.4-19-gcbd2a00/atac_dnase_pipelines.def",
      "high-throughput-sequencing/sve/0.1.0/sve.def",
      "high-throughput-sequencing/sicer/2-1.0.2/sicer.def",
      "high-throughput-sequencing/salmon/1.4.0/salmon.def",
      "high-throughput-sequencing/dropest/0.8.6/dropest.def",
      "high-throughput-sequencing/tandemtools/current/tandemtools.def",
      "high-throughput-sequencing/multiqc/1.9/multiqc.def",
      "high-throughput-sequencing/multiqc/1.10/multiqc.def",
      "high-throughput-sequencing/svtk/0.1/svtk.def",
      "high-throughput-sequencing/ricopili/2019_Jun_25.001/ricopili.def",
      "high-throughput-sequencing/bison/0.4.0/bison.def",
      "high-throughput-sequencing/umitools/1.1.1/umitools.def",
      "high-throughput-sequencing/eager/1.92/eager.def",
      "high-throughput-sequencing/deepvariant/0.10.0/deepvariant.def",
      "high-throughput-sequencing/deepvariant/0.9.0/deepvariant.def",
      "high-throughput-sequencing/deepvariant/1.1.0/deepvariant.def",
      "high-throughput-sequencing/macs/2.2.7.1/macs.def",
      "high-throughput-sequencing/medaka/1.0.3/medaka.def",
      "high-throughput-sequencing/medaka/0.12.1/medaka.def",
      "high-throughput-sequencing/medaka/1.2.0/medaka.def",
      "high-throughput-sequencing/svtyper/0.7.1/svtyper.def",
      "high-throughput-sequencing/scramble/0.0.20190211.82c78b9/scramble.def",
      "high-throughput-sequencing/scramble/1.0.1-32893ef/scramble.def",
      "high-throughput-sequencing/parliament/0.1.7/parliament.def",
      "high-throughput-sequencing/cutadapt/3.0/cutadapt.def",
      "high-throughput-sequencing/cutadapt/2.10/cutadapt.def",
      "high-throughput-sequencing/cutadapt/1.18/cutadapt.def",
      "high-throughput-sequencing/abruijn/1.0/abruijn.def",
      "high-throughput-sequencing/canvas/1.40/canvas.def",
      "high-throughput-sequencing/seqlinkage/1.0/seqlinkage.def",
      "high-throughput-sequencing/gossamer/ac492a8/gossamer.def",
      "high-throughput-sequencing/hicexplorer/3.5.1/hicexplorer.def",
      "high-throughput-sequencing/hicpro/2.11.4/hicpro.def",
      "high-throughput-sequencing/pvactools/2.0.1/pvactools.def",
      "high-throughput-sequencing/pvactools/1.5.5/pvactools.def",
      "high-throughput-sequencing/mtoolbox/1.1/mtoolbox.def",
      "high-throughput-sequencing/bamsurgeon/1111e5d/bamsurgeon.def",
      "high-throughput-sequencing/gridss/2.9.4/gridss.def",
      "high-throughput-sequencing/bigscale2/20191119/bigscale2.def",
      "high-throughput-sequencing/xengsort/28762aac/xengsort.def",
      "high-throughput-sequencing/csvkit/1.0.5/csvkit.def",
      "high-throughput-sequencing/cnvkit/0.9.8/cnvkit.def",
      "high-throughput-sequencing/cnvkit/0.9.6/cnvkit.def",
      "high-throughput-sequencing/fusioninspector/2.5.0/fusioninspector.def",
      "high-throughput-sequencing/fusioninspector/2.3.0/fusioninspector.def",
      "high-throughput-sequencing/megalodon/2.2.9/megalodon.def",
      "high-throughput-sequencing/slamdunk/0.4.3/slamdunk.def",
      "high-throughput-sequencing/rsd/1.1.7/rsd.def",
      "high-throughput-sequencing/tvc/5.10.1/tvc.def",
      "high-throughput-sequencing/maggie/0.3.4/maggie.def",
      "high-throughput-sequencing/flye/2.8-1/flye.def",
      "high-throughput-sequencing/flye/2.7/flye.def",
      "high-throughput-sequencing/tpmcalculator/0.0.4/tpmcalculator.def",
      "high-throughput-sequencing/tpmcalculator/0.0.3/tpmcalculator.def",
      "high-throughput-sequencing/idep/0.81/idep.def",
      "high-throughput-sequencing/transvar/2.5.9/transvar.def",
      "high-throughput-sequencing/flappie/1.0.0/flappie.def",
      "high-throughput-sequencing/flappie/2.1.3/flappie.def",
      "high-throughput-sequencing/cellsnp/0.1.7/cellsnp.def",
      "high-throughput-sequencing/cellsnp/0.3.2/cellsnp.def",
      "high-throughput-sequencing/crossmap/0.5.2/crossmap.def",
      "high-throughput-sequencing/brass/6.3.4/brass.def",
      "high-throughput-sequencing/brass/6.1.2/brass.def",
      "high-throughput-sequencing/rseqc/4.0.0/rseqc.def",
      "high-throughput-sequencing/hail/0.2.61/hail.def",
      "high-throughput-sequencing/hail/0.2.56/hail.def",
      "high-throughput-sequencing/hail/0.2.3/hail.def",
      "high-throughput-sequencing/rmats/4.0.2/rmats.def",
      "high-throughput-sequencing/mitosuite/1.0.9b/mitosuite.def",
      "high-throughput-sequencing/taiji/1.1.0/taiji.def",
      "high-throughput-sequencing/taiji/1.2.0/taiji.def",
      "high-throughput-sequencing/humann2/2.8.1/humann2.def",
      "high-throughput-sequencing/vireosnp/0.5.1/vireosnp.def",
      "high-throughput-sequencing/vireosnp/0.3.2/vireosnp.def",
      "high-throughput-sequencing/cnvnator/0.4.1/cnvnator.def",
      "high-throughput-sequencing/delly/0.8.7/delly.def",
      "high-throughput-sequencing/mageck-vispr/0.5.4/mageck-vispr.def",
      "high-throughput-sequencing/guppy/4.2.2/guppy.def",
      "high-throughput-sequencing/guppy/3.4.5/guppy.def",
      "high-throughput-sequencing/guppy/4.0.15/guppy.def",
      "high-throughput-sequencing/cancerit-wgs/2.1.0/cancerit-wgs.def",
      "high-throughput-sequencing/repeatmodeler/2.0.1/repeatmodeler.def",
      "high-throughput-sequencing/surpi/1.0.67/surpi.def",
      "high-throughput-sequencing/bamliquidator/1.3.8/bamliquidator.def",
      "high-throughput-sequencing/neusomatic/0.2.1/neusomatic.def",
      "high-throughput-sequencing/rnapeg/current/rnapeg.def",
      "high-throughput-sequencing/biom-format/2.1.10/biom-format.def",
      "high-throughput-sequencing/pcap-core/4.3.5/pcap-core.def",
      "high-throughput-sequencing/humann/3.0.0-alpha.3/humann.def",
      "high-throughput-sequencing/crispresso/2.0.40/crispresso.def",
      "high-throughput-sequencing/crispresso/2.0.45/crispresso.def",
      "high-throughput-sequencing/busco/5.0.0/busco.def",
      "high-throughput-sequencing/busco/4.1.3/busco.def",
      "high-throughput-sequencing/bamreadcount/cram-v0.0.1/bamreadcount.def",
      "high-throughput-sequencing/lefse/1.0.8/lefse.def",
      "high-throughput-sequencing/lefse/1.0.7/lefse.def",
      "high-throughput-sequencing/cgpbattenberg/3.5.3/cgpbattenberg.def",
      "high-throughput-sequencing/pychopper/2.4.0/pychopper.def",
      "high-throughput-sequencing/freebayes/1.3.5/freebayes.def",
      "high-throughput-sequencing/raremetal/4.15.1/raremetal.def",
      "high-throughput-sequencing/stream/20180816/stream.def",
      "linkage-phylogenetics/bali-phy/3.5/bali-phy.def",
      "linkage-phylogenetics/gubbins/2.3.4/gubbins.def",
      "structural-biology/parsnip/20180507/parsnip.def",
      "structural-biology/rdock/2013.1/rdock.def",
      "structural-biology/pymol/2.4.0/pymol.def",
      "structural-biology/pymol/2.3.0/pymol_2.3.0.def",
      "deep-learning/basset/0.1.0/basset.def",
      "deep-learning/deeplab/20180816/deeplab.def",
      "deep-learning/caffe2/0.8.1/caffe2.def",
      "deep-learning/dextr-pytorch/20180710/dextr-pytorch.def",
      "deep-learning/tensorrt/18.09/tensorrt.def",
      "deep-learning/clairvoyante/1.0/clairvoyante.def",
      "deep-learning/polyrnnpp/20180718/polyrnnpp.def",
      "deep-learning/few-shot-ssl/20180723/few-shot-ssl.def",
      "deep-learning/digits/6.0/digits.def",
      "deep-learning/unet/20180704/unet.def",
      "mathematical-statistics/omeclust/1.1.4/omeclust.def",
      "mathematical-statistics/omeclust/1.1.6/omeclust.def",
      "mathematical-statistics/m2clust/1.1.3/m2clust.def",
      "mathematical-statistics/m2clust/0.0.7/m2clust.def",
      "mathematical-statistics/m2clust/0.0.8/m2clust.def"
    ],
    "full_name": "NIH-HPC/singularity-def-files",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-nih-hpc-singularity-definition-files\" class=\"anchor\" href=\"#nih-hpc-singularity-definition-files\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eNIH HPC Singularity Definition Files\u003c/h1\u003e\n\u003cp\u003eThese definition files and wrapper scripts are used by the \u003ca href=\"https://hpc.nih.gov/\" rel=\"nofollow\"\u003eNIH HPC (Biowulf)\u003c/a\u003e staff to install containerized applications using \u003ca href=\"https://github.com/sylabs/singularity\"\u003eSingularity\u003c/a\u003e. Each app is installed in a self-contained directory and access to the app is controlled through a module system (\u003ca href=\"https://github.com/TACC/Lmod\"\u003eLmod\u003c/a\u003e). This strategy allows users to transparently access apps that are installed within containers as though they were installed directly on the host system. More details can be found \u003ca href=\"https://hpc.nih.gov/apps/singularity.html#bind-stationary\" rel=\"nofollow\"\u003ehere\u003c/a\u003e.\u003c/p\u003e\n\u003cp\u003eTypically, apps are installed under in a directory structure like so:\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003e$ tree appname/ver\nappname/ver\n|-- bin\n|   |-- cmd1 -\u0026gt; ../libexec/wrapper.sh\n|   |-- cmd2 -\u0026gt; ../libexec/wrapper.sh\n|   `-- cmd3 -\u0026gt; ../libexec/wrapper.sh\n`-- libexec\n    |-- app.sif\n    `-- wrapper.sh\n\u003c/code\u003e\u003c/pre\u003e\n\u003cp\u003eBecause \u003ccode\u003ewrapper.sh\u003c/code\u003e is written to be introspective, any command symlinked to it will be carried through and executed within the associated container. The wrapper script is also sufficiently generic that it can be reused across apps with little or no modification.\u003c/p\u003e\n\u003cp\u003eEach app has its own \u003ccode\u003eREADME.md\u003c/code\u003e that contains:\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003ea link to the NIH HPC app page or developer\u0027s documentation\u003c/li\u003e\n\u003cli\u003ea list of symlinks that should be created to the wrapper script to expose executables within the container\u003c/li\u003e\n\u003cli\u003eany app specific installation notes\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eFinally, please note that these definition files \u003cstrong\u003eare not guaranteed to reproduce the same container, or even to produce any container at all\u003c/strong\u003e. The internet, upon which these definition files are based, is subject to change without notice. These definition files are therefore intended to be treated as (potentially) helpful suggestions.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-computational-chemistry\" class=\"anchor\" href=\"#computational-chemistry\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/computational-chemistry\"\u003eComputational Chemistry\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/computational-chemistry/ampl\"\u003eampl\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-deep-learning\" class=\"anchor\" href=\"#deep-learning\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/deep-learning\"\u003eDeep Learning\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/deep-learning/caffe2\"\u003eCaffe2\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/deep-learning/dextr-pytorch\"\u003eDEXTR-PyTorch\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/deep-learning/polyrnnpp\"\u003ePolyRNNpp\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/deep-learning/basset\"\u003ebasset\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/deep-learning/clairvoyante\"\u003eclairvoyante\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/deep-learning/deeplab\"\u003edeeplab\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/deep-learning/digits\"\u003edigits\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/deep-learning/few-shot-ssl\"\u003efew-shot-ssl\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/deep-learning/tensorrt\"\u003etensorrt\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/deep-learning/unet\"\u003eunet\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-high-throughput-sequencing\" class=\"anchor\" href=\"#high-throughput-sequencing\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/high-throughput-sequencing\"\u003eHigh Throughput Sequencing\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/atac_dnase_pipelines\"\u003eATAC-Seq / DNase-Seq Pipeline\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/ascatngs\"\u003eAscatNGS\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/atropos\"\u003eAtropos\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/bamsurgeon\"\u003eBAMSurgeon\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/brass\"\u003eBRASS\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/canvas\"\u003eCanvas\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/maggie\"\u003eMAGGIE\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/pcap-core\"\u003ePCAP-core\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/pepr\"\u003ePePr\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/rsd\"\u003eRSD\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/surpi\"\u003eSURPI\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/tpmcalculator\"\u003eTPMCalculator\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/tvc\"\u003eTVC\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/vagrent\"\u003eVAGrENT\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/vep\"\u003eVEP\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/abruijn\"\u003eabruijn\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/bamliquidator\"\u003ebamliquidator\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/bamreadcount\"\u003ebamreadcount\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/bamutil\"\u003ebamutil\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/bigscale2\"\u003ebigscale2\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/biom-format\"\u003ebiom-format\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/bison\"\u003ebison\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/busco\"\u003ebusco\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/cancerit-wgs\"\u003ecancerit-wgs\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/cellsnp\"\u003ecellsnp\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/cgpbattenberg\"\u003ecgpBattenberg\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/cicero\"\u003ecicero\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/cnvkit\"\u003ecnvkit\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/cnvnator\"\u003ecnvnator\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/crispresso\"\u003ecrispresso\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/crossmap\"\u003ecrossmap\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/csvkit\"\u003ecsvkit\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/cutadapt\"\u003ecutadapt\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/deepsignal\"\u003edeepsignal\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/deeptools\"\u003edeeptools\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/deepvariant\"\u003edeepvariant\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/delly\"\u003edelly\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/dropest\"\u003edropest\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/eager\"\u003eeager\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/epic2\"\u003eepic2\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/flappie\"\u003eflappie\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/flye\"\u003eflye\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/freebayes\"\u003efreebayes\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/fusioninspector\"\u003efusioninspector\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/gossamer\"\u003egossamer\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/gridss\"\u003egridss\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/guppy\"\u003eguppy\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/hail\"\u003ehail\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/hap.py\"\u003ehap.py\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/hicexplorer\"\u003ehicexplorer\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/hicpro\"\u003ehicpro\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/htseq\"\u003ehtseq\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/humann2\"\u003ehumann2\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/idep\"\u003eidep\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/lefse\"\u003elefse\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/macs\"\u003emacs\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/mageck-vispr\"\u003emageck-vispr\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/medaka\"\u003emedaka\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/megalodon\"\u003emegalodon\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/metabat\"\u003emetabat\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/metaphlan\"\u003emetaphlan\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/mitosuite\"\u003emitosuite\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/mtoolbox\"\u003emtoolbox\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/multiqc\"\u003emultiqc\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/neusomatic\"\u003eneusomatic\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/parliament\"\u003eparliament\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/pvactools\"\u003epvactools\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/pychopper\"\u003epychopper\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/raremetal\"\u003eraremetal\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/repeatmodeler\"\u003erepeatmodeler\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/ricopili\"\u003ericopili\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/rilseq\"\u003erilseq\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/rmats\"\u003ermats\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/rnapeg\"\u003ernapeg\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/rseqc\"\u003erseqc\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/salmon\"\u003esalmon\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/scramble\"\u003escramble\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/seqlinkage\"\u003eseqlinkage\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/sicer\"\u003esicer\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/slamdunk\"\u003eslamdunk\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/stream\"\u003estream\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/sve\"\u003esve\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/svtk\"\u003esvtk\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/svtyper\"\u003esvtyper\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/taiji\"\u003etaiji\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/tandemtools\"\u003etandemtools\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/tetoolkit\"\u003etetoolkit\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/transvar\"\u003etransvar\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/umitools\"\u003eumitools\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/vireosnp\"\u003evireosnp\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/high-throughput-sequencing/xengsort\"\u003exengsort\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-image-analysis\" class=\"anchor\" href=\"#image-analysis\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/image-analysis\"\u003eImage Analysis\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/resmap\"\u003eResMap\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/terastitcher\"\u003eTeraStitcher\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/baracus\"\u003ebaracus\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/broccoli\"\u003ebroccoli\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/civet\"\u003ecivet\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/ctf\"\u003ectf\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/deepmedic\"\u003edeepmedic\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/fitlins\"\u003efitlins\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/fmriprep\"\u003efmriprep\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/minc-toolkit\"\u003eminc-toolkit\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/mriqc\"\u003emriqc\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/mrtrix\"\u003emrtrix\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/qsiprep\"\u003eqsiprep\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/tesseract\"\u003etesseract\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/topaz\"\u003etopaz\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/image-analysis/xcpengine\"\u003excpengine\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-linkage-phylogenetics\" class=\"anchor\" href=\"#linkage-phylogenetics\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/linkage-phylogenetics\"\u003eLinkage Phylogenetics\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/linkage-phylogenetics/bali-phy\"\u003ebali-phy\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/linkage-phylogenetics/gubbins\"\u003egubbins\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-mass-spectrometry\" class=\"anchor\" href=\"#mass-spectrometry\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/mass-spectrometry\"\u003eMass Spectrometry\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/mass-spectrometry/maxquant\"\u003emaxquant\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-mathematicalstatistics\" class=\"anchor\" href=\"#mathematicalstatistics\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/mathematical-statistics\"\u003eMathematical/Statistics\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/mathematical-statistics/m2clust\"\u003em2clust\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/mathematical-statistics/omeclust\"\u003eomeClust\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-molecular-modeling-graphics\" class=\"anchor\" href=\"#molecular-modeling-graphics\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/molecular-modeling-graphics\"\u003eMolecular Modeling Graphics\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/molecular-modeling-graphics/chimerax\"\u003eChimeraX\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/molecular-modeling-graphics/blender\"\u003eblender\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/molecular-modeling-graphics/starseqr\"\u003estarseqr\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-sequence-analysis\" class=\"anchor\" href=\"#sequence-analysis\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/sequence-analysis\"\u003eSequence Analysis\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/acfs\"\u003eACFS\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/annogesic\"\u003eANNOgesic\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/asgal\"\u003eASGAL\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/accurity\"\u003eAccurity\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/eukrep\"\u003eEukRep\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/glu\"\u003eGLU\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/htgtsrep\"\u003eHTGTSrep\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/orffinder\"\u003eORFfinder\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/saige\"\u003eSAIGE\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/vcf-kit\"\u003eVCF-kit\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/anvio\"\u003eanvio\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/arriba\"\u003earriba\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/augustus\"\u003eaugustus\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/bamgineer\"\u003ebamgineer\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/braker\"\u003ebraker\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/cactus\"\u003ecactus\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/chipseq_pipeline\"\u003echipseq_pipeline\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/cicero\"\u003ecicero\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/deepsea\"\u003edeepsea\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/focus\"\u003efocus\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/glnexus\"\u003eglnexus\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/intarna\"\u003eintarna\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/ldsc\"\u003eldsc\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/m-tools\"\u003em-tools\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/mmarge\"\u003emmarge\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/netoglyc\"\u003enetOglyc\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/phaser\"\u003ephaser\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/prokka\"\u003eprokka\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/qtltools\"\u003eqtltools\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/roary\"\u003eroary\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/smoove\"\u003esmoove\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/sonicparanoid\"\u003esonicparanoid\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/svtools\"\u003esvtools\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/wisexome\"\u003ewisexome\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/sequence-analysis/xhla\"\u003exHLA\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-structural-biology\" class=\"anchor\" href=\"#structural-biology\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/structural-biology\"\u003eStructural Biology\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/structural-biology/parsnip\"\u003eparsnip\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/structural-biology/pymol\"\u003epymol\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/structural-biology/rdock\"\u003erDock\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-systems-biology\" class=\"anchor\" href=\"#systems-biology\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/systems-biology\"\u003eSystems Biology\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/systems-biology/cellphonedb\"\u003ecellphonedb\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-utilities\" class=\"anchor\" href=\"#utilities\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003e\u003ca href=\"/utilities\"\u003eUtilities\u003c/a\u003e\n\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/utilities/xvfb\"\u003eXvfb\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/ariba\"\u003eariba\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/atom\"\u003eatom\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/datalad\"\u003edatalad\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/gdc-client\"\u003egdc-client\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/longshot\"\u003elongshot\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/pdf2svg\"\u003epdf2svg\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/pyega3\"\u003epyega3\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/snp-sites\"\u003esnp-sites\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/sysbench\"\u003esysbench\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/uropa\"\u003europa\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/vcf2db\"\u003evcf2db\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/visidata\"\u003evisidata\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/utilities/whatshap\"\u003ewhatshap\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n",
    "stargazers_count": 8,
    "subscribers_count": 7,
    "topics": [],
    "updated_at": 1622586137.0
  },
  {
    "data_format": 2,
    "description": "Scripts for building Singularity images",
    "filenames": [
      "tensorflow/ubuntu.def",
      "mxnet/ubuntu.def",
      "caffe2/ubuntu.def",
      "dl/ubuntu.def",
      "circuitscape/ubuntu.def",
      "caffe/ubuntu.def"
    ],
    "full_name": "clemsonciti/singularity-images",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularity-image-scripts\" class=\"anchor\" href=\"#singularity-image-scripts\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eSingularity image scripts\u003c/h1\u003e\n\u003cp\u003eScripts to generate singularity images\nfor running different software on Palmetto cluster.\u003c/p\u003e\n",
    "stargazers_count": 10,
    "subscribers_count": 5,
    "topics": [],
    "updated_at": 1597386388.0
  },
  {
    "data_format": 2,
    "description": "Intel HPC Containers using Singularity",
    "filenames": [
      "definitionFiles/WRF/wrfRun.def",
      "definitionFiles/WRF/wrfBuild.def",
      "definitionFiles/gromacs/gromacsBuild.def",
      "definitionFiles/gromacs/gromacsRun.def",
      "definitionFiles/namd/namdBuild.def",
      "definitionFiles/namd/namdRun.def",
      "definitionFiles/lammps/lammpsRun.def",
      "definitionFiles/lammps/lammpsBuild.def",
      "definitionFiles/base/base.def"
    ],
    "full_name": "intel/HPC-containers-from-Intel",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-goal\" class=\"anchor\" href=\"#goal\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eGoal:\u003c/h1\u003e\n\u003cp\u003eCreate containers using Singularity definition file for HPC apps and run them on the cloud or bare metal for Single and Cluster runs.\u003c/p\u003e\n\u003cp\u003eThis repo should have definition files only for few HPC applications. Users can utilize them to generate containers.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-get-help\" class=\"anchor\" href=\"#get-help\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eGet Help\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003ca href=\"https://github.com/intel/HPC-containers-from-Intel/issues\"\u003ePost an issue\u003c/a\u003e if you face any problem building or running a container\u003c/li\u003e\n\u003c/ul\u003e\n",
    "stargazers_count": 16,
    "subscribers_count": 9,
    "topics": [
      "hpc",
      "cluster",
      "singularity-containers",
      "cloud"
    ],
    "updated_at": 1619711561.0
  },
  {
    "data_format": 2,
    "description": "Geant4 Example Application with Rich features and Small footprints",
    "filenames": [
      "INSTALL/Singularity"
    ],
    "full_name": "jintonic/gears",
    "latest_release": "v1.4.2",
    "readme": "\u003cp\u003e\u003ca href=\"https://codedocs.xyz/jintonic/gears/annotated.html\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/69e499a9ac348bd92b62e3eac1391967c594141c2855f42783e17f07527553c4/68747470733a2f2f636f6465646f63732e78797a2f6a696e746f6e69632f67656172732e737667\" alt=\"Doxygen\" data-canonical-src=\"https://codedocs.xyz/jintonic/gears.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"https://opensource.org/licenses/MIT\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/78f47a09877ba9d28da1887a93e5c3bc2efb309c1e910eb21135becd2998238a/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f4c6963656e73652d4d49542d79656c6c6f772e737667\" alt=\"License\" data-canonical-src=\"https://img.shields.io/badge/License-MIT-yellow.svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"examples\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/11e2b1b7847229fdbb3430cf4a5d908964287c64b59e0119936bce5359e4efb9/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f67656172732d6578616d706c65732d626c75653f7374796c653d666c6174\" alt=\"Examples\" data-canonical-src=\"https://img.shields.io/badge/gears-examples-blue?style=flat\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"INSTALL\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/04a8d93a22dbf7fc5499c5c388432c900fac24a8e4061f7794fa5c2260199c46/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f6765742d737461727465642d6f72616e67653f7374796c653d666c6174\" alt=\"Get Started\" data-canonical-src=\"https://img.shields.io/badge/get-started-orange?style=flat\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"#how-to-contribute\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/b02b9d3708d187a71e102cc95c8a8ad5fcdffb8241ba9b4a102a4272862f1216/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f6765742d696e766f6c7665642d6666363962343f7374796c653d666c6174\" alt=\"Get Involved\" data-canonical-src=\"https://img.shields.io/badge/get-involved-ff69b4?style=flat\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"examples/detector/visualization/gearsX3D.html\"\u003e\u003cimg align=\"right\" width=\"120px\" src=\"examples/detector/visualization/gears.png\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003e\u003ca href=\"https://github.com/jintonic/gears\"\u003eGEARS\u003c/a\u003e is a \u003ca href=\"http://geant4.cern.ch\" rel=\"nofollow\"\u003eGeant4\u003c/a\u003e \u003ca href=\"http://geant4-userdoc.web.cern.ch/geant4-userdoc/UsersGuides/ForApplicationDeveloper/html/Examples/examples.html\" rel=\"nofollow\"\u003eExample\u003c/a\u003e Application with \u003ca href=\"#features\"\u003eRich features\u003c/a\u003e yet Small footprint. The entire C++ coding is minimized down to a single file with about 550 \u003ca href=\"https://en.wikipedia.org/wiki/Source_lines_of_code\" rel=\"nofollow\"\u003eSLOC\u003c/a\u003e. This is achieved mainly by utilizing \u003ca href=\"http://geant4.cern.ch\" rel=\"nofollow\"\u003eGeant4\u003c/a\u003e plain \u003ca href=\"http://geant4-userdoc.web.cern.ch/geant4-userdoc/UsersGuides/ForApplicationDeveloper/html/Detector/Geometry/geomASCII.html\" rel=\"nofollow\"\u003etext geometry description\u003c/a\u003e, \u003ca href=\"http://geant4-userdoc.web.cern.ch/geant4-userdoc/UsersGuides/ForApplicationDeveloper/html/Control/commands.html\" rel=\"nofollow\"\u003ebuilt-in UI commands\u003c/a\u003e (macros), and C++ inheritance. It is ideal for student training and fast implementation of small to medium-sized experiments.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-features\" class=\"anchor\" href=\"#features\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eFeatures\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003ca href=\"gears.cc\"\u003eSingle small C++ file\u003c/a\u003e, easy to manage, fast to \u003ca href=\"INSTALL#compile-gears\"\u003ecompile\u003c/a\u003e (a few second on a regular PC)\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"examples/physics\"\u003eEasy switching between well maintained Geant4 reference physics lists without recompilation\u003c/a\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"examples/physics#physics-processes\"\u003eIndividual processes can be turned on/off without recompilation\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"examples/physics#optical-properties-of-materials-and-surfaces\"\u003eFast implementation of optical properties without recompilation\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"examples/physics#radioactive-decay\"\u003eOptional radioactive decay simulation\u003c/a\u003e with the possibility to \u003ca href=\"examples/physics#split-decay-chain\"\u003esave the parent and daughter decays into different events if the later happens after a user specified time interval\u003c/a\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"examples/sources#common-sources\"\u003eFrequently used source spectra (AmBe, Am-241, etc.)\u003c/a\u003e in addition to \u003ca href=\"examples/sources\"\u003eGPS\u003c/a\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"examples/output\"\u003eOutput in multiple data format\u003c/a\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003ca href=\"examples/output#root\"\u003eROOT\u003c/a\u003e TTree format (default, no \u003ca href=\"https://root.cern.ch\" rel=\"nofollow\"\u003eROOT\u003c/a\u003e installation is needed)\n\u003cul\u003e\n\u003cli\u003eBuild-in data compression, well suitable for large data processing\u003c/li\u003e\n\u003cli\u003eFast access to independent data members\u003c/li\u003e\n\u003cli\u003eFlat tree (no nested branches or arrays) with short leaf names\n\u003cul\u003e\n\u003cli\u003eEasy to use in TTree::Draw\u003c/li\u003e\n\u003cli\u003eNo need to load extra library to open\u003c/li\u003e\n\u003cli\u003eCan be easily analyzed in \u003ca href=\"https://www.python.org/\" rel=\"nofollow\"\u003ePython\u003c/a\u003e through \u003ca href=\"https://github.com/scikit-hep/uproot4\"\u003eUproot\u003c/a\u003e.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"https://www.hdfgroup.org/downloads/hdf5/\" rel=\"nofollow\"\u003eHDF5\u003c/a\u003e, universal data format, easy to read by different tools\u003c/li\u003e\n\u003cli\u003eCSV or XML, Human readable ASCII file, capable of dealing with multiple dimensional arrays\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"examples/output#record-information-of-step-0\"\u003eRecord information of step 0\u003c/a\u003e (initStep), which is not available through \u003ca href=\"http://www-geant4.kek.jp/lxr/source/tracking/include/G4UserSteppingAction.hh\" rel=\"nofollow\"\u003eG4UserSteppingAction\u003c/a\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"http://geant4-userdoc.web.cern.ch/geant4-userdoc/UsersGuides/ForApplicationDeveloper/html/Detector/Geometry/geomASCII.html\" rel=\"nofollow\"\u003esimple text\u003c/a\u003e or \u003ca href=\"https://gdml.web.cern.ch/GDML/\" rel=\"nofollow\"\u003eGDML\u003c/a\u003e geometry I/O\n\u003cul\u003e\n\u003cli\u003e\n\u003ca href=\"examples/detector\"\u003eFast implementation of detector geometry\u003c/a\u003e without C++ programming\u003c/li\u003e\n\u003cli\u003eCreate/Change geometry without re-compilation\u003c/li\u003e\n\u003cli\u003eTurn off data saving in a volume by assigning it a non-positive copy number\u003c/li\u003e\n\u003cli\u003eTurn any volume to a \u003ca href=\"examples/detector#sensitive-volume\"\u003esensitive detector\u003c/a\u003e by adding \"(S)\" in its name\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"examples/detector/optical\"\u003eAssign optical properties in Geant4 plain text geometry description\u003c/a\u003e, which is not available in the official \u003ca href=\"http://geant4.cern.ch\" rel=\"nofollow\"\u003eGeant4\u003c/a\u003e release\u003c/li\u003e\n\u003cli\u003e\n\u003ca href=\"examples/detector/syntax\"\u003eSyntax highlighting of the simple text geometry description files\u003c/a\u003e in \u003ca href=\"examples/detector/syntax#emacs\"\u003eEmacs\u003c/a\u003e, \u003ca href=\"examples/detector/syntax#vim\"\u003eVim\u003c/a\u003e, \u003ca href=\"examples/detector/syntax#micro\"\u003eMicro\u003c/a\u003e, and \u003ca href=\"examples/detector/syntax#sublime-text\"\u003eSublime Text\u003c/a\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"http://geant4-userdoc.web.cern.ch/geant4-userdoc/UsersGuides/ForApplicationDeveloper/html/Detector/commandScore.html\" rel=\"nofollow\"\u003eCreating 3D mesh to record and visualize physical variables in it without any change of the C++ code\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://codedocs.xyz/jintonic/gears/\" rel=\"nofollow\"\u003eDoxygen documentation\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eMany \u003ca href=\"examples\"\u003esample macros\u003c/a\u003e and \u003ca href=\"examples/detector\"\u003egeometry descriptions\u003c/a\u003e for feature demonstration\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-how-to-contribute\" class=\"anchor\" href=\"#how-to-contribute\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eHow to contribute\u003c/h2\u003e\n\u003cp\u003ePlease \u003ca href=\"https://help.github.com/en/github/getting-started-with-github/fork-a-repo\"\u003efork GEARS on GitHub\u003c/a\u003e. Run the following to get a local copy of the forked repository and link it to the \u003ca href=\"https://github.com/jintonic/gears\"\u003eoriginal GEARS repository\u003c/a\u003e:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003e$ git clone git@github.com:yourGitHubAccount/gears.git \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e get forked repository\u003c/span\u003e\n$ git remote add upstream git@github.com:jintonic/gears.git \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e link to original repository\u003c/span\u003e\n$ git remote -v \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e run a check\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eRun the following to keep your local repository updated with the \u003ca href=\"https://github.com/jintonic/gears\"\u003eoriginal GEARS repository\u003c/a\u003e:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003e$ git fetch upstream \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e updates are saved in a new branch upstream/master\u003c/span\u003e\n$ git merge upstream/master \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e merge 2 branches: upstream/master and master\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp\u003eIf the merge is successful, run \u003ccode\u003egit push\u003c/code\u003e to update your forked GEARS repository on GitHub.\u003c/p\u003e\n\u003cp\u003eYou can initiate a \u003ca href=\"https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests\"\u003epull request on GitHub\u003c/a\u003e if you\u0027d like to have your update being absorbed in \u003ca href=\"https://github.com/jintonic/gears\"\u003ethe original GEARS repository\u003c/a\u003e.\u003c/p\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-coding-convention\" class=\"anchor\" href=\"#coding-convention\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eCoding convention\u003c/h3\u003e\n\u003ch4\u003e\n\u003ca id=\"user-content-g4cout-vs-stdcout\" class=\"anchor\" href=\"#g4cout-vs-stdcout\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eG4cout VS std::cout\u003c/h4\u003e\n\u003cp\u003e\u003ccode\u003eG4cout\u003c/code\u003e and \u003ccode\u003eG4endl\u003c/code\u003e is preferred over \u003ccode\u003estd:cout\u003c/code\u003e and \u003ccode\u003estd:endl\u003c/code\u003e because the former handle the output in \u003ca href=\"http://geant4.cern.ch\" rel=\"nofollow\"\u003eGeant4\u003c/a\u003e GUI correctly, while the later can only output to terminal.\u003c/p\u003e\n\u003ch4\u003e\n\u003ca id=\"user-content-indentation\" class=\"anchor\" href=\"#indentation\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eIndentation\u003c/h4\u003e\n\u003cp\u003eTwo spaces instead of a tab are used to indent a line in \u003ca href=\"gears.cc\"\u003egears.cc\u003c/a\u003e to insure a consistent appearance in different text editors, and to avoid wasting space in front of deeply nested code blocks. The following mode lines are added to the end of \u003ca href=\"gears.cc\"\u003egears.cc\u003c/a\u003e to insure that in \u003ca href=\"https://www.vim.org/\" rel=\"nofollow\"\u003eVim\u003c/a\u003e and \u003ca href=\"https://www.gnu.org/software/emacs/\" rel=\"nofollow\"\u003eEmacs\u003c/a\u003e:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-c++\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e//\u003c/span\u003e -*- C++; indent-tabs-mode:nil; tab-width:2 -*-\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e//\u003c/span\u003e vim: ft=cpp:ts=2:sts=2:sw=2:et\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003ch3\u003e\n\u003ca id=\"user-content-to-dos\" class=\"anchor\" href=\"#to-dos\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eTo-do\u0027s\u003c/h3\u003e\n\u003cul\u003e\n\u003cli\u003eexamples\n\u003cul\u003e\n\u003cli\u003eadd an example to show how QE can be implemented\u003c/li\u003e\n\u003cli\u003eadd examples to show how one can distribute source in a volume or surface\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n",
    "stargazers_count": 22,
    "subscribers_count": 10,
    "topics": [
      "geant4",
      "detector",
      "physics",
      "monte-carlo-simulation"
    ],
    "updated_at": 1622745498.0
  },
  {
    "data_format": 2,
    "description": "A repository of definition files for bootstrapping Singularity containers around the software applications, frameworks, and libraries you need to run on high-performance computing systems.",
    "filenames": [
      "definition-files/tensorflow/Singularity.tensorflow-2.3.0",
      "definition-files/ciml/Singularity.tape-0.4",
      "definition-files/ciml/Singularity.esm-0.3.1",
      "definition-files/omb/Singularity.omb-5.6.3-ubuntu-18.04-openmpi-3.1.6",
      "definition-files/omb/Singularity.omb-5.7-ubuntu-18.04-cuda-11.2-openmpi-4.0.5",
      "definition-files/omb/Singularity.omb-5.7-ubuntu-18.04-openmpi-4.0.5",
      "definition-files/omb/Singularity.omb-5.6.3-ubuntu-18.04-cuda-10.1.168-openmpi-3.1.4",
      "definition-files/omb/Singularity.omb-5.6.3-ubuntu-18.04-mvapich-2.3.2",
      "definition-files/omb/Singularity.omb-5.6.3-ubuntu-18.04-openmpi-3.1.4",
      "definition-files/omb/Singularity.omb-5.6.3-centos-7.9.2009-openmpi-3.1.4",
      "definition-files/omb/Singularity.omb-5.6.3-centos-7.9.2009-mvapich-2.3.2",
      "definition-files/mxnet/Singularity.mxnet-1.7.0",
      "definition-files/xcrysden/Singularity.xcrysden-1.6.2",
      "definition-files/gromacs/Singularity.gromacs-2020.3-ubuntu-18.04-cuda-10.1.168-tmpi-avx-512-cuda-70",
      "definition-files/fenics/Singularity.fenics-2019.1.0",
      "definition-files/paraview/Singularity.paraview-5.8.1-ubuntu-18.04-openmpi-3.1.6-osmesa-20.1.5",
      "definition-files/paraview/Singularity.paraview-5.9.0-ubuntu-18.04-openmpi-3.1.6-osmesa-20.1.5",
      "definition-files/paraview/Singularity.paraview-5.8.1-ubuntu-18.04-openmpi-3.1.4-osmesa-20.1.5",
      "definition-files/miniconda/Singularity.miniconda2-py27-4.8.3-ubuntu-18.04",
      "definition-files/miniconda/Singularity.miniconda3-py37-4.9.2-ubuntu-18.04",
      "definition-files/ior/Singularity.ior-3.3.0rc1-ubuntu-18.04-openmpi-3.1.6",
      "definition-files/ior/Singularity.ior-3.3.0rc1-ubuntu-18.04-openmpi-3.1.4",
      "definition-files/ior/Singularity.ior-3.3.0-ubuntu-18.04-openmpi-4.0.5",
      "definition-files/pytorch/Singularity.pytorch-1.5.0",
      "definition-files/spark/Singularity.spark-3.1.2-hadoop-3.2-ubuntu-18.04",
      "definition-files/spark/Singularity.spark-2.3.1-hadoop-2.7-ubuntu-18.04",
      "definition-files/hpl/Singularity.hpl-2.3-ubuntu-18.04-openmpi-3.1.4-openblas-0.3.10",
      "definition-files/hpl/Singularity.hpl-2.3-ubuntu-18.04-openmpi-4.0.5-openblas-0.3.14",
      "definition-files/hpl/Singularity.hpl-2.3-ubuntu-18.04-openmpi-3.1.6-openblas-0.3.10",
      "definition-files/ubuntu/Singularity.ubuntu-18.04-cuda-10.1.168",
      "definition-files/ubuntu/Singularity.ubuntu-18.04-mvapich-2.3.2",
      "definition-files/ubuntu/Singularity.ubuntu-18.04-openmpi-4.0.5",
      "definition-files/ubuntu/Singularity.ubuntu-18.04-openmpi-3.1.4",
      "definition-files/ubuntu/Singularity.ubuntu-20.04",
      "definition-files/ubuntu/Singularity.ubuntu-18.04-cuda-10.1.168-openmpi-3.1.4",
      "definition-files/ubuntu/Singularity.ubuntu-18.04-cuda-11.2",
      "definition-files/ubuntu/Singularity.ubuntu-18.04-cuda-10.2",
      "definition-files/ubuntu/Singularity.ubuntu-18.04-cuda-10.2-openmpi-3.1.6",
      "definition-files/ubuntu/Singularity.ubuntu-18.04-openmpi-3.1.6",
      "definition-files/ubuntu/Singularity.ubuntu-18.04",
      "definition-files/ubuntu/Singularity.ubuntu-18.04-cuda-11.2-openmpi-4.0.5",
      "definition-files/visit/Singularity.visit-3.1.4-ubuntu-18.04-openmpi-3.1.6",
      "definition-files/centos/Singularity.centos-7.9.2009-cuda-10.1.168",
      "definition-files/centos/Singularity.centos-7.9.2009-openmpi-3.1.4",
      "definition-files/centos/Singularity.centos-7.9.2009-mvapich-2.3.2",
      "definition-files/centos/Singularity.centos-7.9.2009",
      "definition-files/beast/Singularity.beast-2.6.1",
      "definition-files/beast/Singularity.beast-1.10.4",
      "definition-files/anaconda/Singularity.anaconda3-py38-2020.11-ubuntu-18.04",
      "definition-files/anaconda/Singularity.anaconda2-py27-2019.10-ubuntu-18.04",
      "definition-files/stream/Singularity.stream-5.10-ubuntu-18.04",
      "definition-files/singularity/Singularity.singularity-3.5.3"
    ],
    "full_name": "mkandes/naked-singularity",
    "latest_release": null,
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-naked-singularity\" class=\"anchor\" href=\"#naked-singularity\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003enaked-singularity\u003c/h1\u003e\n\u003cp\u003eA repository of definition files for building\n\u003ca href=\"https://sylabs.io/guides/latest/user-guide\" rel=\"nofollow\"\u003eSingularity\u003c/a\u003e containers\naround the software applications, frameworks, and libraries you need to\nrun on high-performance computing systems.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-install-singularity\" class=\"anchor\" href=\"#install-singularity\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eInstall Singularity\u003c/h2\u003e\n\u003cp\u003eInstall Singularity on your Linux desktop, laptop, or virtual machine.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003e./naked-singularity.sh install\u003c/pre\u003e\u003c/div\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-build-a-singularity-container-from-a-definition-file\" class=\"anchor\" href=\"#build-a-singularity-container-from-a-definition-file\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eBuild a Singularity container from a definition file\u003c/h2\u003e\n\u003cp\u003eBuild an Ubuntu Singularity container from one of the definition files\navailable in this repository.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003esudo singularity build ubuntu.sif definition-files/ubuntu/Singularity.ubuntu-18.04\u003c/pre\u003e\u003c/div\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-download-an-existing-singularity-container\" class=\"anchor\" href=\"#download-an-existing-singularity-container\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eDownload an existing Singularity container\u003c/h2\u003e\n\u003cp\u003eA number of pre-built containers from this repository are also now\nhosted at Singularity Hub.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell\"\u003e\u003cpre\u003esingularity pull shub://mkandes/naked-singularity:ubuntu-18.04\u003c/pre\u003e\u003c/div\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-status\" class=\"anchor\" href=\"#status\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eStatus\u003c/h2\u003e\n\u003cp\u003eA work in progress. But an important note on the immediate future ...\u003c/p\u003e\n\u003cp\u003eThe future availablity of Singularity Hub is currently unknown and\nmay be slated for retirement as early as April 2021. If and when\nSingularity Hub is retired, the current set of naked-sinularity\ndefinition files will be affected due to the recent move in the past\nyear to using a layered dependency chain of multiple containers built\nand hosted on Singularity Hub. Alternative container build and hosting\noptions are currently under consideration. This upcoming change will\nlikely affect how a given container\u0027s dependency chain is rebuilt and\nsupported long-term.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-contribute\" class=\"anchor\" href=\"#contribute\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eContribute\u003c/h2\u003e\n\u003cp\u003eIf you would like to contribute one of your own Singularity container\ndefinition files for a specific application OR request a modification to\nan existing container definition, then please submit a pull request.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-author\" class=\"anchor\" href=\"#author\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eAuthor\u003c/h2\u003e\n\u003cp\u003eMarty Kandes, Ph.D.\u003cbr\u003e\nComputational \u0026amp; Data Science Research Specialist\u003cbr\u003e\nHigh-Performance Computing User Services Group\u003cbr\u003e\nSan Diego Supercomputer Center\u003cbr\u003e\nUniversity of California, San Diego\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-version\" class=\"anchor\" href=\"#version\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eVersion\u003c/h2\u003e\n\u003cp\u003e1.3.3\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-last-updated\" class=\"anchor\" href=\"#last-updated\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eLast Updated\u003c/h2\u003e\n\u003cp\u003eSaturday, March 13th, 2021\u003c/p\u003e\n",
    "stargazers_count": 31,
    "subscribers_count": 2,
    "topics": [],
    "updated_at": 1622849740.0
  },
  {
    "data_format": 2,
    "description": "SingularityCE is the Community Edition of Singularity, an open source container platform designed to be simple, fast, and secure.",
    "filenames": [
      "examples/legacy/2.2/arch.def",
      "examples/legacy/2.2/scientific.def",
      "examples/legacy/2.2/busybox.def",
      "examples/legacy/2.2/ubuntu.def",
      "examples/legacy/2.2/debian.def",
      "examples/legacy/2.2/docker.def",
      "examples/legacy/2.2/centos.def",
      "examples/legacy/2.2/contrib/ubuntu-bio.def",
      "examples/legacy/2.2/contrib/debian85-tensorflow-0.10.def",
      "examples/legacy/2.2/contrib/linuxbrew_and_non-root_software_example.def",
      "examples/legacy/2.2/contrib/ubuntu16-tensorflow-0.12.1.def",
      "examples/legacy/2.2/contrib/ubuntu-openfoam.def",
      "examples/legacy/2.2/contrib/centos7-ompi_master.def",
      "examples/legacy/2.2/contrib/centos-minimal.def",
      "examples/legacy/2.2/contrib/r_python_julia.def",
      "examples/legacy/2.2/contrib/centos7-ompi_cuda.def",
      "examples/legacy/2.2/contrib/ubuntu16-tensorflow-0.12.1-gpu.def",
      "examples/legacy/2.2/contrib/ubuntu-root.def",
      "examples/legacy/2.2/contrib/fedora.def",
      "examples/legacy/2.3/contrib/raspbian.def",
      "examples/build-singularity/build-singularity.def",
      "e2e/testdata/inspecter_container.def",
      "e2e/testdata/Docker_registry.def",
      "e2e/testdata/sshfs.def",
      "e2e/testdata/regressions/issue_5315.def",
      "e2e/testdata/regressions/issue_4203.def",
      "e2e/testdata/regressions/issue_4583.def",
      "e2e/testdata/regressions/issue_4969.def",
      "e2e/testdata/regressions/issue_5399.def",
      "e2e/testdata/regressions/issue_5250.def",
      "e2e/testdata/regressions/issue_4820.def",
      "e2e/testdata/regressions/issue_4967.def"
    ],
    "full_name": "sylabs/singularity",
    "latest_release": "v3.8.0",
    "readme": "\u003ch1\u003e\n\u003ca id=\"user-content-singularityce\" class=\"anchor\" href=\"#singularityce\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eSingularityCE\u003c/h1\u003e\n\u003cp\u003e\u003ca href=\"https://circleci.com/gh/sylabs/singularity/tree/master\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/ff56e7dd170e08e53c09fda12031315bb91f5b4220f2d3cfaf46044700f32fa1/68747470733a2f2f636972636c6563692e636f6d2f67682f73796c6162732f73696e67756c61726974792f747265652f6d61737465722e7376673f7374796c653d737667\" alt=\"CircleCI\" data-canonical-src=\"https://circleci.com/gh/sylabs/singularity/tree/master.svg?style=svg\" style=\"max-width:100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"CONTRIBUTING.md\"\u003eGuidelines for Contributing\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\".github/PULL_REQUEST_TEMPLATE.md\"\u003ePull Request Template\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"LICENSE.md\"\u003eProject License\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://www.sylabs.io/docs/\" rel=\"nofollow\"\u003eDocumentation\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#support\"\u003eSupport\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#citing-singularity\"\u003eCitation\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eSingularityCE is the Community Edition of Singularity, an open source container\nplatform designed to be simple, fast, and secure. Singularity is optimized\nfor compute focused enterprise and HPC workloads, allowing untrusted users\nto run untrusted containers in a trusted way.\u003c/p\u003e\n\u003cp\u003eCheck out \u003ca href=\"https://www.sylabs.io/videos\" rel=\"nofollow\"\u003etalks about Singularity\u003c/a\u003e and some \u003ca href=\"https://sylabs.io/case-studies\" rel=\"nofollow\"\u003euse\ncases of Singularity\u003c/a\u003e on our website.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-getting-started-with-singularityce\" class=\"anchor\" href=\"#getting-started-with-singularityce\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eGetting Started with SingularityCE\u003c/h2\u003e\n\u003cp\u003eTo install SingularityCE from source, see the \u003ca href=\"INSTALL.md\"\u003einstallation\ninstructions\u003c/a\u003e. For other installation options, see \u003ca href=\"https://www.sylabs.io/guides/latest/admin-guide/\" rel=\"nofollow\"\u003eour\nguide\u003c/a\u003e.\u003c/p\u003e\n\u003cp\u003eSystem administrators can learn how to configure SingularityCE, and get an\noverview of its architecture and security features in the \u003ca href=\"https://www.sylabs.io/guides/latest/admin-guide/\" rel=\"nofollow\"\u003eadministrator\nguide\u003c/a\u003e.\u003c/p\u003e\n\u003cp\u003eFor users, see the \u003ca href=\"https://www.sylabs.io/guides/latest/user-guide/\" rel=\"nofollow\"\u003euser\nguide\u003c/a\u003e for details on how to use\nand build Singularity containers.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-contributing-to-singularityce\" class=\"anchor\" href=\"#contributing-to-singularityce\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eContributing to SingularityCE\u003c/h2\u003e\n\u003cp\u003eCommunity contributions are always greatly appreciated. To start developing\nSingularityCE, check out the \u003ca href=\"CONTRIBUTING.md\"\u003eguidelines for contributing\u003c/a\u003e.\u003c/p\u003e\n\u003cp\u003eWe also welcome contributions to our \u003ca href=\"https://github.com/sylabs/singularity-userdocs\"\u003euser\nguide\u003c/a\u003e and \u003ca href=\"https://github.com/sylabs/singularity-admindocs\"\u003eadmin\nguide\u003c/a\u003e.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-support\" class=\"anchor\" href=\"#support\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eSupport\u003c/h2\u003e\n\u003cp\u003eTo get help with SingularityCE, check out the community spaces\ndetailed at our \u003ca href=\"https://www.sylabs.io/singularity/community/\" rel=\"nofollow\"\u003eCommunity\nPortal\u003c/a\u003e.\u003c/p\u003e\n\u003cp\u003eSee also our \u003ca href=\"SUPPORT.md\"\u003eSupport Guidelines\u003c/a\u003e for further\ninformation about the best place, and how, to raise different kinds of\nissues and questions.\u003c/p\u003e\n\u003cp\u003eFor additional support, \u003ca href=\"https://www.sylabs.io/contact/\" rel=\"nofollow\"\u003econtact us\u003c/a\u003e to receive\nmore information.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-citing-singularity\" class=\"anchor\" href=\"#citing-singularity\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eCiting Singularity\u003c/h2\u003e\n\u003cpre\u003e\u003ccode\u003eKurtzer GM, Sochat V, Bauer MW (2017) Singularity: Scientific containers for mobility of compute. PLoS ONE 12(5): e0177459. https://doi.org/10.1371/journal.pone.0177459\n\u003c/code\u003e\u003c/pre\u003e\n\u003cp\u003eWe also have a Zenodo citation:\u003c/p\u003e\n\u003cpre\u003e\u003ccode\u003eKurtzer, Gregory M. et. al. Singularity - Linux application and environment\ncontainers for science. 10.5281/zenodo.1310023\n\u003c/code\u003e\u003c/pre\u003e\n\u003cp\u003e\u003ca href=\"https://doi.org/10.5281/zenodo.1310023\" rel=\"nofollow\"\u003ehttps://doi.org/10.5281/zenodo.1310023\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eThis is an \u0027all versions\u0027 DOI. Follow the link to Zenodo to obtain a DOI specific\nto a particular version of Singularity.\u003c/p\u003e\n\u003ch2\u003e\n\u003ca id=\"user-content-license\" class=\"anchor\" href=\"#license\" aria-hidden=\"true\"\u003e\u003cspan aria-hidden=\"true\" class=\"octicon octicon-link\"\u003e\u003c/span\u003e\u003c/a\u003eLicense\u003c/h2\u003e\n\u003cp\u003e\u003cem\u003eUnless otherwise noted, this project is licensed under a 3-clause BSD license\nfound in the \u003ca href=\"LICENSE.md\"\u003elicense file\u003c/a\u003e.\u003c/em\u003e\u003c/p\u003e\n",
    "stargazers_count": 34,
    "subscribers_count": 7,
    "topics": [
      "containers",
      "hpc",
      "linux"
    ],
    "updated_at": 1622788883.0
  }
]
